---
title: "Phylotranscriptomics reveals discordance in the phylogeny of Hawaiian _Drosophila_ and _Scaptomyza_ (Diptera: Drosophilidae)"
output:
  bookdown::pdf_document2:
  keep_tex: yes
  latex_engine: pdflatex
  #bookdown::html_document2: default
toc: false
bibliography: Hawaiian_Drosophila_phylogeny.bib
csl: nature.csl
link-citations: yes
header-includes:
 \usepackage{xr} \externaldocument{Hawaiian_Drosophila_phylogeny_supplement}
 \usepackage{float}
 \usepackage{array}
 \usepackage{xcolor,colortbl}
 \floatplacement{figure}{H}
---

```{r global options, include = FALSE}
knitr::opts_chunk$set(echo=FALSE, include = FALSE, warning=FALSE, message=FALSE, cache=FALSE)

load("Hawaiian_Drosophila_phylogeny_supplement.RData")
library(dplyr)
```

Samuel H. Church^1^\*, Cassandra G. Extavour^1,2^

^1^ Department of Organismic and Evolutionary Biology, Harvard University, Cambridge, MA 02138, USA

^2^ Department of Molecular and Cellular Biology, Harvard University, Cambridge, MA 02138, USA

\* Corresponding author

# Abstract {-}

Island radiations present natural laboratories for studying the evolutionary process. The Hawaiian Drosophilidae are one such radiation, with nearly 600 described species and substantial morphological and ecological diversification. These species are largely divided into a few major clades, but the relationship between these clades remains uncertain. Here we present 12 new assembled transcriptomes from across these clades, and use these transcriptomes to resolve the base of the evolutionary radiation. We recover a new hypothesis for the relationship between clades, and demonstrate its support over previously published hypotheses. We then use the evolutionary radiation to explore dynamics of concordance in phylogenetic support, by analyzing the gene and site concordance factors for every possible topological combination of major groups. We show that high bootstrap values mask low evolutionary concordance, and we demonstrate that the most likely topology is distinct from the topology with the highest support across gene trees and from the topology with highest support across sites. We then combine all previously published genetic data for the group to estimate a time-calibrated tree for over 300 species of drosophilids. Finally, we digitize dozens of published Hawaiian Drosophilidae descriptions, and use this to pinpoint probable evolutionary shifts in reproductive ecology as well as body, wing, and egg size. We show that by examining the entire landscape of tree and trait space, we can gain a more complete understanding of how evolutionary dynamics play out across an island radiation.

# Introduction {-}

In the era of genome-scale data, we have an opportunity to unpack the biological meaning of phylogenetic support. In phylogenetic analyses that seek to discover the relationships between organisms, support is often defined as the proportion of information that favors a particular branch in an evolutionary tree[@simon2020evolving]. Methods have been developed that emphasize extracting the tree with the greatest amount of support from out of an otherwise rugged landscape of treespace[@swofford1996phylogenetic;@hedtke2006resolution]. However, a growing number of studies have emphasized the biological relevance of that landscape to our understanding of the evolutionary process[@kellogg1996genes;@baum2007concordance;@smith2015analysis]. For example, many new studies have contributed evidence that, even with trees with high measures of conventional support, we can expect large amounts of discordance among sites and genes, especially when examining speciation events with short internodes or with a likelihood of introgression[@pease2016phylogenomics;@weisrock2012concatenation]. Here we use the island radiation of Hawaiian drosophilid flies to study the landscape of treespace, and show that the relationships between the major groups of these flies are best understood by using methods that embrace evolutionary discordance.

The Hawaiian _Drosophila_ have a long history as a model clade for the implementation of phylogenetic methods[@o2018hawaiian]. More than twenty years ago, Baker and Desalle used the Hawaiian radiation of _Drosophila_ to perform one of the first analyses to demonstrate incongruence between an overall species tree and underlying gene trees[@baker1997multiple]. Their study focused on the resolution between major clades of Hawaiian _Drosophila_ and built on the landmark work done by Carson in the 1970s inferring the phylogeny of a subgroup of Hawaiian _Drosophila_, the picture-wing flies, based on the banding pattern of polytene chromosomes[@carson1976drosophila], among other early phylogenetic studies[@beverley1985ancient;@thomas1991molecular]. During the past twenty years, the relationships between major groups has been revisited several times[@Bonacum2001;@kambysellis1995pattern]. Most recently, O'Grady and colleagues (2011) used mitochondrial genes and expanded taxon sampling[@o2011phylogenetic], and Magnacca and Price (2015) used an expanded nuclear gene set [@magnacca2015rapid]. The study presented here builds on this foundational work, presenting the first phylogenetic analysis of genome-scale data for the group.

The Hawaiian Drosophilidae consist of 566 described species[@nudidrosophila2010taxonomic;@magnacca2012new], with hundreds more estimated to be awaiting description[@nudidrosophila2010taxonomic]. These species have been divided into the following major clades[@nudidrosophila2010taxonomic]: [1] the _picture-wing_, _nudidrosophila_, _ateledrosophila_ (PNA) clade, which has served as a model clade for the study of sexual selection[@kaneshiro1987sexual] and speciation[@kang2016genomic]; [2] the _antopocerus_, _modified-tarsus_, _ciliated-tarsus_ (AMC) clade, first proposed by Heed (1968),[@heed1968ecology;@nudidrosophila2010taxonomic] and confirmed by subsequent phylogenetic studies [@o2011phylogenetic;@lapoint2014phylogenetics]; [3] the _modified-mouthparts_ (MM) clade; and [4] the _haleakalae_ clade, an enigmatic group in need of further study[@hardy2001review]. Several other smaller clades have been suggested as falling outside of these major groups, including the _rustica_ group of three species[@o2001rustica], and the monotypic lineages of _D. primaeva_ and _D. adventitia_. The position of _D. primaeva_ has been somewhat uncertain, but several studies have suggested it is the sister taxon to _picture-wing_ flies[@Bonacum2001], including the work on polytene chromosomes by Carson and Stalker[@carson1969polytene]. The species _D. adventitia_ was originally suggested to be part of the MM clade[@hardy1965insects], but recent studies placed it as the sister taxon to _D. primaeva_[@Bonacum2001] or possibly other major clades. Additionally, the Hawaiian _Drosophila_ are the sister clade of the genus _Scaptomyza_, which is nested within the broader paraphyletic genus _Drosophila_ and is hypothesized to have colonized the island independently[@throckmorton1966relationships;@lapoint2013diversification], possibly more than once[@katoh2017multiple]. Throughout this manuscript, we use Hawaiian _Drosophila_ to refer to non-_Scaptomyza_ Hawaiian species, and Hawaiian Drosophilidae to refer to the clade of Hawaiian _Drosophila_+_Scaptomyza_.

Many phylogenetic studies have been performed which have confirmed the monophyly of each of these clades and provided resolution for internal relationships (PNA[@bonacum2005phylogeny;@magnacca2015rapid], AMC[@lapoint2014phylogenetics;@lapoint2011phylogenetic], _haleakalae_[@o2004phylogenetic], and _Scaptomyza_[@lapoint2013diversification;@katoh2017multiple]). Previous phylogenetic studies, however, have not resulted in a consensus relationship between the major clades within Hawaiian _Drosophila_ (Fig. \ref{fig:previous-phylogenetic-hypotheses})[@magnacca2015rapid]. Magnacca and Price (2015) showed that different phylogenetic methods of analysis (e.g. using software based on Bayesian statistics rather than maximum likelihood for inference) produced highly incongruent topologies (Fig. 1)[@magnacca2015rapid]. In that study, the most likely topology had _D. primaeva_ as the sister taxon to all other Hawaiian _Drosophila_, and included a clade uniting MM+AMC+_haleakalae_, with the _haleakalae_ clade showing greater affinity to AMC species relative to MM species (Fig. 1B). This topology was consistent with the tree suggested by O'Grady and colleagues (2011) analysing mitochondrial data and using maximum likelihood and Bayesian analyses[@o2011phylogenetic]. However the analyses of Magnacca and Price (2015) using Bayesian software package BEAST showed an alternative relationship, with _haleakalae_ flies as the sister clade to all other Hawaiian _Drosophila_, a clade uniting the MM+PNA+_D. primaeva_, and an closer affinity between _D. primaeva_ and PNA species than between _D. primaeva_ and MM species (Fig. 1C). This latter arrangement is largely consistent with relationships proposed by Throckmorton in 1966[@throckmorton1966relationships] and reiterated in several subsequent studies (Fig. \ref{fig:previous-phylogenetic-hypotheses})[@baker1997multiple;@Bonacum2001;@kambysellis1995pattern].

Resolving these relationships is critical for our understanding of the morphological and ecological evolution of these flies[@o2011phylogenetic;@Bonacum2001;@kambysellis1995pattern]. Hawaiian _Drosophila_ demonstrate a large diversity in body size[@stevenson1995organ], wing size[@edwards2007database], and egg size[@montague1981reproductive]; in the number and position of structural features such as wing spots[@edwards2007database]; in the number of egg-producing units in the ovary (ovarioles)[@kambysellis1971studies;@sarikaya2019reproductive]; and in the type of substrate used for oviposition and larval feeding[@kambysellis1995pattern;@magnacca2008review]. Some clades demonstrate unique suites of morphological and behavioral traits, whose evolutionary history is unclear because of uncertainties in the phylogeny. For example, the _haleakalae_ flies exclusively use fungal oviposition substrates and are considered to have less complex mating behaviors than other, more well-studied groups (e.g. _picture-wing_ flies)[@hardy2001review]. It is unclear whether this suite of traits represents a secondary transition relative to the ancestral state, because it is not known whether _haleakalae_ flies are the sister clade to all other Hawaiian _Drosophila_ or nested within the radiation. Resolution in the relationships at the base of this lineage will be key in identifying which branches experienced substantial trait diversification, and especially in identifying whether any of these traits demonstrate predictable patterns of co-evolution.

Here we present the first phylogenomic relationships between the major groups of Hawaiian Drosophilidae. We combine twelve new transcriptomes sequenced in this study with recently published genomes for two Hawaiian _Drosophila_ species[@kim2020highly], four non-Hawaiian _Scaptomyza_[@kim2020highly], and six outgroup species[@larkin2021flybase]. By increasing the number of genes used to infer relationships, we begin to unpack the evolutionary history in the short internodes at the base of the Hawaiian _Drosophila_ radiation. Following up on the critical study by Baker and Desalle 25 years ago[@baker1997multiple], we explore the landscape of treespace and the discordance between species and gene trees using our phylotranscriptomic dataset. We then use the results of our analysis as initial constraints on subsequent phylogenetic analyses using a dataset of `r ape::Ntip(root_all_tree)` species and `r length(gene_file_names)` genes, compiled using all previous phylogenetic studies of Hawaiian Drosophilidae. Finally, we estimate the age of the radiation, and use this time-calibrated tree to identify branches where shifts in trait evolution likely occurred. Our findings suggest a relationship between major clades that is distinct from both previously hypothesized topologies, and that is well supported by both maximum likelihood and Bayesian analyses. We show that examining a comprehensive landscape of tree and trait space can allow for a more complete understanding of evolutionary dynamics in this remarkable island radiation.

# Methods {-}

## Field collection and RNA extraction {-}

### Field collection {-}

Specimens used for transcriptome sampling were caught on the Hawaiian islands between May of 2016 and May of 2017. Specimens were caught using a combination of net sweeping and fermented banana-mushroom baits in various field sites on the Hawaiian islands of Kaua'i and Hawai'i (see Supplemental Table \ref{tab:collection-info} for locality data). Field collections were performed under permits issued by the following: Hawai'i Department of Land and Natural Resources, Hawai'i Island Forest Reserves, Kaua'i Island Forest Reserves, Koke'e State Park, and Hawai'i Volcanoes National Park. Adult flies were maintained in the field on vials with a sugar-based media and kept at cool temperatures. They were transported alive back to Cambridge, MA where they were maintained on standard _Drosophila_ media at 18&deg;C. Samples were processed for RNA extraction between `r min(collection_info$'days elapsed')` and `r max(collection_info$'days elapsed')` days after collecting them live in the field (average `r round(mean(collection_info$'days elapsed'),1)` days, see Supplemental Table \ref{tab:collection-info}). One species, _Scaptomyza varia_, was caught in the field before the adult stage by sampling rotting _Clermontia sp._ flowers (the oviposition substrate). For this species, male and female adult flies emerged in the lab, and were kept together until sampled for RNA extraction.

### Species identification {-}

Species were identified using dichotomous keys[@magnacca2012new;@hardy1977review;@hackman1959genus;@o2003revision;@hardy1965insects] when possible. Many keys for Hawaiian Drosophilidae are written focusing on adult male specific characters (e.g. sexually dimorphic features or male genitalia)[@hackman1959genus]. Therefore, for species where females could not be unambiguously identified, we verified their identity using DNA barcoding. When males were caught from the same location, we identified males to species using dichotomous keys and matched their barcode sequences to females included in our study. When males were not available, we matched barcodes from collected females to sequences previously uploaded to NCBI[@o2011phylogenetic;@lapoint2014phylogenetics;@katoh2017multiple].

The following dichotomous keys were used to identify species: for _picture-wing_ males and females, Magnacca and Price (2012)[@magnacca2012new]; for _antopocerus_ males, Hardy (1977)[@hardy1977review]; for _Scaptomyza_, Hackman (1959)[@hackman1959genus]; for species in the _mimica_ subgroup of MM, O'Grady and colleagues (2003)[@o2003revision]; for other miscellaneous species, Hardy (1965)[@hardy1965insects].

For DNA barcoding, DNA was extracted from one or two legs from male specimens using the Qiagen DNeasy blood and tissue extraction kit, or from the DNA of females isolated during RNA extraction (see below). We amplified and sequenced the cytochrome oxidase I (COI), II (COII) and 16S rRNA genes using the primers and protocols described in Sarikaya and colleagues (2019)[@sarikaya2019reproductive]. 

For barcode matching, we aligned sequences using MAFFT, version v7.475[@katoh2013mafft], and assembled gene trees using RAxML, version 8.2.9[@stamatakis2014raxml]. Definitive matches were considered when sequences for females formed a monophyletic clade with reference males or reference sequences from NCBI (Supplemental Table \ref{tab:barcode-info}). Sequence files and gene trees are available at the GitHub repository http://github.com/shchurch/hawaiian_drosophilidae_phylogeny_2021, under `analysis/data/DNA_barcoding`.

Female _D. primaeva_, _D. macrothrix_, _D. sproati_, and _D. picticornis_ could be identified unambiguously using dichotomous keys. Female _D. atroscutellata_, _D. nanella_, _D. mimica_, _D. tanythrix_, _S. cyrtandrae_, _S. varipicta_, and _S. varia_ were identified by matching barcodes to reference sequences from NCBI, reference males, or both. For the female _haleakalae_ fly used in this study, no male flies were caught in the same location as these individuals, and no other sequences for _haleakalae_ males on NCBI were an exact match with this species. Given its similar appearance to _Drosophila dives_, we are referring to it here as _Drosophila_ cf _dives_, and we await further molecular and taxonomic studies of this group that will resolve its identity. Photos of individual flies used for transcriptome sequencing are shown in Fig. \ref{fig:voucher-photos}.
 
### RNA extraction {-}

RNA was extracted from frozen samples using the standard TRIzol protocol (http://tools.thermofisher.com/content/sfs/manuals/trizol_reagent.pdf). One mL of TRIzol was added to each frozen sample, which was then homogenized using a sterile motorized mortar. The recommended protocol was followed without modifications, using 10 µg of glycogen, and resuspending in 20µL RNAse-free water-EDTA-SDS solution. DNA for subsequent barcoding was also extracted using the phenol-chloroform phase saved from the RNA extraction. 

RNA concentration was checked using a Qubit fluorometer, and integrity was assessed with an Agilent TapeStation 4200. RNA libraries were prepared following the PrepX polyA mRNA Isolation kit and the PrepX RNA-Seq for Illumina Library kit, using the 48 sample protocol on an Apollo 324 liquid handling robot in the Harvard University Bauer Core Facilities. Final library concentration and integrity were again assessed using the Qubit and TapeStation protocols.

The field collecting for this study was accomplished with a target number of individuals per species in mind, based on future sampling objectives for RNA sequencing studies that, as of the time of writing, have not been published. These objectives were to have four wild-caught, mature, apparently healthy females, three of which were to be dissected for tissue-specific RNA sequencing, and one intended as a whole body reference library. When four individuals were not available, the reference library was assembled by combining the tissue specific libraries from one of the other individuals. This was the case for the following species: _D. sproati_, which was dissected and had RNA extracted separately from the head, ovaries, and carcass, with RNA combined prior to library preparation; and _S. varia_, _S. cyrtandrae_ and _D._ cf _dives_, for which RNA was extracted and libraries prepared for separate tissues, and raw reads were combined after sequencing. 

For the other eight species, sufficient individual females were available such that reads for transcriptome assembly were sequenced from a separate individual. In these cases one entire female fly was dissected and photographed to assess whether vitellogenic eggs were present in the ovary, and all tissues were combined in the same tube and used for RNA extraction. 

Libraries for transcriptome assembly were sequenced on an Illumina HiSeq 2500, using the standard version 4 protocol, at 125 base pairs of paired-end reads. A table of total read counts for each library can be found in Supplemental Table \ref{tab:reads-info}. 

## Transcriptome assembly {-}

Transcriptome assembly was performed using the agalma pipeline, version 2.0.0[@dunn2013agalma]. For the 12 new transcriptomes presented in this study, reads from separate rounds of sequencing were concatenated and inserted into the agalma catalog. These were combined with seven publicly available outgroup genomes (_D. virilis_, _D. mojavensis_, _D. pseudoobscura_, _D. ananassae_, _D. willistoni_, and _D. melanogaster_[@larkin2021flybase]), two Hawaiian genomes (_D. grimshawi_[@larkin2021flybase] and _D. murphyi_[@kim2020highly]), and four _Scaptomyza_ genomes (_S. graminum_, _S. montana_, _S. hsui_, _S pallida_[@kim2020highly]). For the non-Hawaiian _Drosophila_ and _D. grimshawi_ genomes, the longest isoform per gene was selected using the gene header. For the four _Scaptomyza_ genomes and _D. murphyi_ genomes, single copy orthologs were filtered using BUSCO version 4.1.4[@seppey2019busco] against the Diptera obd10 gene set (over 98% of genes were retained as single-copy orthologs). See Supplemental Table \ref{tab:genome-info} for genome information.  

Using the agalma pipeline, the quality score of each library was assessed, and transcriptomes were assembled using the standard parameters. The publicly available genomes were translated and annotated, and the homology of assembled products was inferred using the all-by-all blast component of the `homologize` step in the agalma pipeline, using nucleotide data and a GTR+Gamma model to infer gene trees. The agalma version 2.0.0 pipeline also performs a step to reduce the effects of transcript misassignment using a phylogenetically informed approach (`treeinform`)[@guang2021revising]. Gene orthology was inferred according to the topology of gene trees estimated with RAxML, orthologs were aligned and trimmed using MAFFT[@katoh2013mafft] and Gblocks[@castresana2000selection] respectively, and a supermatrix of aligned orthologous sequences was exported.

The final supermatrix output from agalma consisted of 10,949 putatively orthologous genes and 12,758,237 sites. For the primary analyses performed in this manuscript, this supermatrix was not filtered by occupancy, and the actual gene occupancy was 41.9% across the 24 species present in this study. We also created a supermatrix filtered to a target occupancy of 80%, which consisted of 1,926
 genes and 1,943,000
 sites, which we used to reestimate the maximum likelihood phylogeny. 

All commands used to run the agalma pipeline, and all output report files, are available at the GitHub repository http://github.com/shchurch/hawaiian_drosophilidae_phylogeny_2021, under `analysis/phylotranscriptomics`. 

## Phylotranscriptomics and concordance factors {-}

We estimated the maximum likelihood phylogeny using IQtree, version 2.1.1[@minh2020iq]. We ran IQtree on a dataset partitioned by transcripts, and using the default Model Finder[@kalyaanamoorthy2017modelfinder] per partition[@chernomor2016terrace]. For this analysis, partitions containing no informative sites were excluded. We estimated 1000 ultrafast bootstraps[@hoang2018ufboot2] for each node. We also used IQtree to estimate the gene and site concordance factors, as described in Minh and colleagues (2020)[@minh2020new]. We ran this analysis first on a concatenated dataset output by the agalma pipeline command `supermatrix` with no gene occupancy threshold (returning all aligned transcripts), and then repeated it on a matrix with an 80% occupancy threshold. All subsequent phylogenetic analyses were performed on the full dataset.

We also estimated the maximum likelihood phylogeny using the `speciestree` step of the agalma pipeline, which itself runs RAxML, version 8.2.9[@stamatakis2014raxml]. We used the default parameters for RAxML as called within the agalma phylotranscriptomic pipeline (model GTR+Gamma, 1000 bootstraps).

We compared the most likely tree against two alternative hypotheses (Fig. 1B-C) using the Swofford-Olsen-Waddell-Hillis (SOWH) test[@swofford1996phylogenetic], as implemented in sowhat, version 1.0[@church2015automation]. We ran both comparisons using a GTR+Gamma model, unpartitioned data file, and 100 simulated datasets.

We estimated the phylogeny using the Bayesian software PhyloBayes, mpi version 1.7a[@lartillot2013phylobayes]. We ran PhyloBayes using a CAT-GTR model for nucleotide data, without partitions, on the full set of transcripts exported from agalma. Phylobayes was run for 1400-1900 generations, and convergence was assessed as the maximum difference between two chains. The initial two chains did not show signs of convergence after 1000 generations (maximum difference was 1), so two additional chains were initialized. These reached convergence with one of the initial chains at 450 generations (maximum difference was 0). The divergence between these three chains and the fourth resulted from differences in the relationships between the MM, AMC, and _haleakalae_ clades. The consensus tree was estimated using all four chains and burn-in of 100 generations, taking every tree (maximum difference was 1).

We estimated the phylogeny using the multi-species coalescent with the software ASTRAL, version 5.7.7[@zhang2018astral]. For this analysis we input the gene trees as inferred by IQtree, using the methods described above. 

To further explore the concordance of data across possible topologies in treespace, we wrote a custom python script to create all `r length(comb_trees)` combinations of possible topologies for the five clades in question, with the root between these clades set at the split between Hawaiian _Drosophila_ and _Scaptomyza_. We used each of these trees as the constraint for a concordance factor analysis, using the same approach as described above for the most likely tree. We visualized treespace by plotting each tree according to Robinsoun-Foulds distance using the R package treespace, version 1.1.4[@jombart2017treespace]. We then mapped on this space the mean concordance factors for each topology (calculated as the mean site and gene concordance on branches, excluding those shared between all topologies).

All commands used to execute the concordance factor analysis are included in the GitHub repository http://github.com/shchurch/hawaiian_drosophilidae_phylogeny_2021 in the rmarkdown file for the supplement of this manuscript, as well as the folder `analysis/phylotranscriptomics/concordance-factor`.

## Estimating an expanded phylogeny {-}

We used the phylotranscriptomic results above, combined with previously published genetic data for Hawaiian Drosophilidae, to estimate an expanded phylogeny. First we gathered the accession numbers from all previously published studies of Hawaiian Drosophila and Scaptomyza genetics[@baker1997multiple;@o2004phylogenetic;@bonacum2005phylogeny;@lapoint2011phylogenetic;@lapoint2013diversification;@lapoint2013diversification;@lapoint2014phylogenetics;@magnacca2015rapid;@katoh2017multiple]. Nucleotide data for each accession number were downloaded from NCBI in March of 2019. We made no manual alterations to these sequences, with the following exceptions: We replaced all non-nucleotide sites (e.g. 'N', 'R') with missing data ('?'); we removed two sequences (U94256.1 - _D. disjuncta_ and U94262.1 - _S. albovittata_) from the 16S dataset that did not align to other sequences; we manually removed a portion of the COI dataset that did not align; we corrected spelling for _S. albovittata_; and we updated the taxonomic name of _D. crassifemur_ to _S. crassifemur_. Original and modified sequences are provided at the GitHub repository http://github.com/shchurch/hawaiian_drosophilidae_phylogeny_2021 under `analysis/time-calibrated_phylogenetics/downloaded_sequences`.

We then aligned these sequences using MAFFT, version 7.457[@katoh2013mafft], `--auto` option. We visualized alignments, and for gene 16S we repeated the alignment using the `--adjustdirectionaccurately` option. We removed all information from the headers with the exception of the species name, and then selected the sequence per species with the fewest gaps. We concatenated sequences using phyutility version 2.2.6[@smith2008phyutility]. 

Using these concatenated sequences, we estimated a phylogeny for `r length(all_tree_tips)` species, including `r length(described_tips)` described species and `r length(undescribed_tips)` that are undescribed but for which genetic vouchers were available. This tree was estimated using IQtree[@minh2020iq] with the topology constrained using the most likely phylotranscriptomic tree. This constraint tree included only taxa overlapping between the phylotranscriptomic and concatenated datasets, with one exception: _D. iki_ was substituted for _D._ cf _dives_, given that this unidentified species was the only representative from the _haleakalae_ clade present in the phylotranscriptomic analysis. No partition model was used for this analysis. We ran IQtree with default parameters, and we estimated 1000 ultrafast bootstrap support values as well as 1000 SH-like likelihood ratio tests. 

Visualizing the results showed that all major clades (AMC, PNA, MM, _haleakalae_, and _Scaptomyza_) were recovered as monophyletic, with the exception of the placement of _D. konaensis_, a member of the hirtitibia subgroup that was recovered as the sister taxon to the AMC clade. We investigated the source of this discrepancy by analyzing the individual gene trees that had representation for this species (COI, COII, and 16S, tree estimated using IQtree as described above, tree files available at `analysis/time-calibrated_phylogenetics/iqtree/iqtree_investigations`). These gene trees showed that _D. konaensis_ sequences had variable affinity to unlikely relatives, including _Scaptomyza_ and _modified-mouthpart_. We considered this to be an artifact of a possible error in accession sequence, and so we removed _D. konaensis_ from downloaded sequences and repeated the alignment and tree estimation steps.

All commands used to download and align sequences as well as estimate the phylogeny, along with all input and output files, are available at the GitHub repository http://github.com/shchurch/hawaiian_drosophilidae_phylogeny_2021, under `analysis/time-calibrated_phylogenetics/`.

## Calibrating the phylogeny to time {-}

This expanded phylogeny was calibrated to time using BEAST, version 2.6.3[@bouckaert2014beast]. This tree search was performed using the following parameters and priors, set using BEAUTi2[@bilderbeek2018babette]: a relaxed log-normal clock model, a general time reversible (GTR) site model with 4 gamma categories, a Yule process branching model, and four normally distributed node priors, based on those used in Magnacca and Price (2015)[@magnacca2015rapid]. These calibrations are based on the apparent progression rule seen in these island distribution of these clades. We adjusted the mean values for island ages to correspond to recently updated estimates for the age at which islands became habitable[@lim2017true], which are based on models that describe the volcanic growth and decay of each Hawaiian island as it has passed over the tectonic hotspot. The mean and sigma values for these calibrations were as follows: mean 4.135 million years, sigma 0.500 for the _planitibia_ and _lanaiensis_ subgroups; mean 2.550, sigma 0.300 for the split between _D. orthofascia_, _D. sobrina_, and _D. ciliatrus_; and mean 1.200, sigma 0.200 for the split between _D. silvestris_ and _D. heteroneura_. We also repeated this analysis using the same mean island ages as recorded in Magnacca and Price (2015)[@magnacca2015rapid].

For all BEAST analysis, the most likely topology from the expanded IQtree search was used to create a starting tree, rooted at the split between _Scaptomyza_ and _Drosophila_ and with branch lengths removed. This topology was fixed throughout the analysis by setting tree topology operator weights to zero.

BEAST analyses were run for between 2 and 2.5 million generations. The maximum clade credibility tree was determined using TreeAnnotator[@helfrich2018treeannotator], with a burn-in of 10\%, selected by visualizing in Tracer, version 1.7.1[@rambaut2018posterior]. The effective size for the posterior was >100 for both analyses (older island ages = `r beast_post_eff` and younger island ages = `r beast_altpost_eff`), while for tree height the effective size for the older island ages was slightly below 100 (older island ages = `r beast_tree_eff` and younger island ages = `r beast_alttree_eff`).

## Estimating ecological and morphological evolutionary transitions {-}

For ecological data on oviposition and larval feeding substrate, we used the rearing records summarized in Magnacca and colleagues (2008)[@magnacca2008review], Appendix I. Following the method of Magnacca _et al_., we grouped oviposition substrates into several general categories, listed in Supplemental Table \ref{tab:substrate-categories}. We also followed the definition from Magnacca and colleagues of non-monophagous (here referred to as generalist) as any species for which no single substrate type comprised more than $\frac{2}{3}$ of rearing records, or for which any two substrates each comprised more than $\frac{1}{4}$. We note that _D. comatifemora_ was listed as a bark breeder in Sarikaya and colleagues (2019)[@sarikaya2019reproductive], but no rearing records for this species are present in Magnacca and colleagues 2008[@magnacca2008review] and Magnacca and O'Grady (2009)[@magnacca2009revision] list it as "breeding habits unknown".

We reconstructed the ancestral state for general oviposition substrate type using the R package phytools, version 0.7-70[@revell2012phytools] on the maximum clade credibility tree from the constrained BEAST analyses. We performed 1000 simulations of stochastic character mapping using the `make.simmap` function (with a maximum likelihood method for estimating the transition model), and then summarized the ancestral state at each node as the oviposition substrate with the highest posterior probability. We used this summary tree to identify branches with likely transitions between oviposition substrates.

For morphological data on wing, body, and thorax length, we digitized data from  26 publications[@grimshaw1901fauna;@grimshaw1902fauna;@knab1914drosophilidae;@bryan1934review;@bryan1938key;@wirth1952two;@hackman1959genus;@hardy1965insects;@hardy1966viii;@hardy1968new;@hardy1969descriptions;@hardy1970notes;@hardy1971new;@hardy1972new;@hardy1977review;@hardy1979review;@kambysellis1971studies;@hardy1975studies;@o2001rustica;@hardy2001review;@o2003revision;@starmer2003phylogenetic;@magnacca2008revision;@magnacca2009revision;@craddock2016hawaiian;@sarikaya2019reproductive]. For data on ovariole number, egg width, and egg length, we digitized data from three publications[@starmer2003phylogenetic;@kambysellis1971studies;@sarikaya2019reproductive]. We made the following modifications to morphological data: In the data from the GitHub repository associated with the study by Sarikaya and colleagues (2019) [@sarikaya2019reproductive], egg size was measured using the radius rather than the diameter; therefore for consistency across studies, we multiplied the reported egg measurements by two. We also excluded data on the egg size of _D. incognita_ from the same publication[@sarikaya2019reproductive] which had two measurements that showed significantly more variation than other measurements (~150% discrepancy in egg length). We excluded the data on wing and thorax length from O'Grady and colleagues (2003)[@o2003revision] for which all measurements were more than three times longer than measurements for conspecific species in other manuscripts.

We identified shifts in evolutionary regimes using the R packages bayou, version 2.2.0[@uyeda2014novel] and SURFACE, version 0.5[@ingram2013surface] on the maximum clade credibility tree from the constrained BEAST analyses. For all analyses, trait data were log~10~ transformed. For species that had multiple records for the same trait across publications, we randomly selected one description (data on intraspecific variation or measurement error were not included in analyses due to inconsistency in the methods used to gather and report these data by the original authors). The bayou analyses were performed using a half-Cauchy distribution for the prior value of alpha and sigma^2^ (scale set at 0.1), a conditional Poisson distribution for the number of shifts (lambda of 10, max of 50), and a normal distribution for theta values (prior mean and standard deviation set at the mean and standard deviation of the trait data). These analyses were run for one million generations, with the exception of body and thorax length, which were run for two million generations. A burn-in value was set at `r burnin` and convergence was evaluated using effective size of the likelihood and the number of shifts (k), see Supplemental Table \ref{tab:bayou-effective-size}. The SURFACE analyses were performed on a combination of egg volume, ovariole number, and body length using default parameters and using a two-step forward-backward process of selecting the number of regimes[@ingram2013surface].

## Data availability {-}

All data, code, and results are available at the GitHub repository http://github.com/shchurch/hawaiian_drosophilidae_phylogeny_2021, commit b12cbb10. This code was implemented in a clean computational environment, which can be recapitulated by following the document `phylo_conda_environment.sh`. Code to reproduce the figures and text for these mansucript files are available as rmarkdown documents. Concordance value results for each of the possible topological arrangements are available at the above GitHub repository. Raw RNA sequencing data are available at the Sequence Read Archive of NCBI, under BioProject PRJNA731506. Assembled transcriptomes and DNA barcode sequences are available at the above GitHub repository.

# Results {-}

## Phylotranscriptomics suggest a new phylogeny of Hawaiian Drosophilidae {-}

Using a phylotranscriptomic approach, we recovered a new topology between the major clades of Hawaiian Drosophilidae, distinct from those previously hypothesized (Fig. 1, \ref{fig:previous-phylogenetic-hypotheses}). This topology was the most likely tree estimated using IQtree[@minh2020iq] and RAxML[@stamatakis2014raxml], as well as the consensus tree with highest posterior probability estimated using PhyloBayes[@lartillot2013phylobayes] (Fig. 1A, \ref{fig:raxml-tree}, \ref{fig:phylobayes-tree}). Bootstrap support for all branches was 100 and posterior probability was 1, with the exception of the branch subtending the clade uniting MM+AMC (IQtree ultrafast bootstrap: 66, RAxML bootstrap: 57, PhyloBayes posterior probability: 0.52). We also estimated the phylogeny using a multi-species coalescent model with ASTRAL[@zhang2018astral], and recovered the same topology with the exception of the placement of _D. primaeva_ (as the sister taxon to PNA, Fig. \ref{fig:astral-tree}). Each of these analyses were performed on a supermatrix of 10,949 putatively orthologous genes, aligned and assembled using the agalma pipeline[@dunn2013agalma] with no filtering based on occupancy (actual gene occupancy was 41.7%). To test the senstivity of our results to missing data, we repeated the IQtree analysis on a dataset reduced using an occupancy threshold that ensures representation of 80% of taxa at each gene (1,926 genes), and recovered the same topology as with the full set of genes (Fig. \ref{fig:graphic-iqtree-occupancy}).

The most likely tree indicates that the PNA clade, including _picture-wing_ species, is the sister clade to all other Hawaiian _Drosophila_. _D. primaeva_ is found to be the sister taxon to a clade containing non-PNA Hawaiian _Drosophila_, though this clade received lower support when using the dataset reduced by occupancy (Fig. \ref{fig:graphic-iqtree-occupancy},  ultrafast bootstrap of 85). A second monotypic lineage, _D. adventitia_, was not sampled for phylotranscriptomic analyses, but using specific gene markers, we recover this as the sister taxon to a clade including AMC+MM+_haleakalae_ (see section on expanded phylogenetic analysis below). This latter clade was previously recovered in previous phylogenetic analyses[@o2011phylogenetic;@magnacca2015rapid]. In contrast to those studies, which suggested a monophyletic clade of AMC+_haleakalae_, we do not recover sufficient support for any particular arrangement of MM, AMC, and _haleakalae_ (ultrafast bootstrap from both the full and reduced occupancy matrix is <95).

We tested the most likely tree emerging from our analysis (Fig. 1A) against two previously suggested alternative hypotheses (Fig. 1B-C) using the Swofford-Olsen-Waddell-Hillis (SOWH) test[@swofford1996phylogenetic], a parametric bootstrap approach for comparing phylogenetic hypotheses. In both cases, the difference in likelihood between the most likely tree and these alternatives was larger than we would expect by chance (p-value for both <0.01, with a sample size of 100). Between Fig. 1A and 1B the difference in log-likelihood was 1774.1, and between Fig. 1A and 1C was 6132.1, while the null distribution according to the SOWH test had no differences greater than 15 for either comparison. Taken together, our results suggest a new phylogeny for Hawaiian Drosophilidae relationships wherein MM, AMC, and _haleakalae_ represent a monophyletic group, and the PNA clade, rather than either the _haleakalae_ clade or _D. primaeva_, is the sister clade to all others (Fig. 1A).

![**Phylotranscriptomic analysis indicates relationships between major clades distinct from those previously hypothesized.**  Photos show six of the twelve species with _de novo_ transcriptomes presented in this study, listing their parent clade and the Hawaiian island on which they are found. A, Results novel to this study, showing best supported tree across maximum likelihood and Bayesian analyses. Node labels indicate ultrafast bootstrap values / gene tree concordance factors (gCF) / site concordance factors (sCF), see concordance factor analysis below.  _D. adventitia_ was not present in phylotranscriptomic analyses, see Fig. 3 for information on its placement. B-C, Previously hypothesized relationships between the _picture wing_-_nudidrosophila_-_ateledrosophila_ (PNA), _modified-mouthparts_ (MM), _antopocerus_-_modified tarsus_-_ciliated tarsus_ (AMC), _haleakalae_, and _Scaptomyza_ clades, as well as two monotypic clades, _D. primaeva_ and _D. adventitia_. Topology B was recovered in O'Grady and colleagues (2011)[@o2011phylogenetic] and Magnacca and Price (2015)[@magnacca2015rapid]. Topology C was recovered using the Bayesian software BEAST in Magnacca and Price (2015)[@magnacca2015rapid], showing incongruent relationships between clades at the base of the radiation of Hawaiian _Drosophila_.](figure_panels/Figure1_V6-01.png)

## Identifying hotspots of gene and site concordance in treespace {-}

We analyzed the strength of phylogenetic concordance in our phylotranscriptomic dataset by estimating the gene and site concordance factors for each branch in our tree. Gene concordance factors (gCF) are calculated as the proportion of informative gene trees that contain a given branch between taxa, and can range from 0 to 100[@minh2020new;@baum2007concordance]. Site concordance factors (sCF) are calculated as the average proportion of informative sites that support a given branch between taxa. Because one site can only support one of three arrangements for a quartet of taxa, sCF typically ranges from ~33.3 to 100, with 33.3 representing our null expectation based on chance[@minh2020new]. We found that for many branches in our tree both gCF and sCF are high, indicating these relationships are supported by a majority of gene and sites in our dataset. For example, the branch uniting Hawaiian _Drosophila_ has a gCF of 91.2, and sCF of 72.1 (Fig. 1A). However for the branches subtending most relationships between the major clades of Hawaiian _Drosophila_, gCF and sCF are low. For example, the branch uniting _D. primaeva_+_haleakalae_+AMC+MM to the exclusion of PNA has a bootstrap value of 100, but a gCF of 19.3 and sCF of 32.2.

We interpret this discordance as reflecting real variation in the phylogenetic signal of different genes and sites, which is not unexpected for a radiation such as this with short internodes subtending major clades[@minh2020new]. Furthermore, the presence of discordance does not mean that there is little that can be said about the relationships between these groups. In contrast, by unpacking this discordance we can begin to qualitatively describe the amount and distribution of phylogenetic signal for multiple alternative, plausible bipartitions.

To this end, we first visualized hotspots of concordance across treespace (Fig. 2). We created all 105 topological combinations of the possible arrangements between major clades, and then re-estimated gCF and sCF for each. Visualizing the mean values for gCF and sCF plotted in treespace shows that the most likely tree, as estimated with IQtree, is not the tree with the highest mean gCF and sCF, but it is near a hotspot of alternative arrangements for which both of these values are high (Fig 2, treespace, most likely tree indicated by dark red outline). In contrast to the most likely topology, the trees with the top three mean gCF values and two of the three trees with the top mean sCF values unite _D. primaeva_+PNA to the exclusion of other Hawaiian _Drosophila_. Variation between these top trees largely depends on the placement of _haleakalae_ relative to other clades (Fig. 2, top gCF and sCF trees).

The mean gCF and sCF across branches may not always be informative metrics, given that some topologies may contain one highly supported branch and others with very low support. Therefore, we also analyzed concordance for all the unique bipartitions across the set of possible topologies (Figs. \ref{fig:graphic-branch-gCF} and \ref{fig:graphic-branch-sCF}, see Supplemental Section 'Concordance Factor Analysis'). We found that for gCF, there is clear signal supporting bipartitions that unite _D. primaeva_+PNA, as well as those that unite MM+AMC+_haleakalae_ (Fig. \ref{fig:graphic-branch-gCF}). We found that for sCF, concordance values across bipartitions are more variable, but those that unite PNA+_haleakalae_ show less support than we might expect by chance, while those that unite _D. primaeva_+PNA and AMC+MM show more support (Fig. \ref{fig:graphic-branch-sCF}). In addition, between gCF and sCF, we found conflicting signals for bipartitions that define one clade as sister to the rest of Hawaiian _Drosophila_, with gCF indicating support for PNA (consistent with the most likely topology), and sCF indicating support for _haleakalae_.

In summary, across all analyses we found strong evidence for a bipartition that separates PNA from clades that include MM and AMC. While the placement of _D. primaeva_ was strongly supported in our maximum likelihood and Bayesian analyses, we found evidence for substantial discordance in this arrangement, and detect signal suggesting a significant amount of shared history between _D. primaeva_ and PNA. Similarly, while the clade uniting MM+AMC+_haleakalae_ received strong bootstrap support, we observed substantial discordance in the placement of _haleakalae_, and suggest that further resolution in its placement will be possible with additional taxon sampling in that clade.

![**The landscape of treespace shows hotpots of concordance among genes and sites.** The landscape of treespace for all possible topological combinations of the five clades of Hawaiian _Drosophila_ studied here: PNA, _D. primaeva_, _haleakalae_, MM, and AMC. Individuals points represent different arrangements of the five clades, labeled randomly with two-letter IDs from aa through ea. The distance between points indicates tree similarity (calculated from Robinson-Foulds distances). The size of points represents the mean gene concordance factor (gCF) across relevant branches, and the color represents the mean site concordance factor (sCF, purple=low, yellow=high). The point outlined in red (tree dm) indicates the best topology found with IQtree, RAxML, and PhyloBayes, which is distinct from the top trees according to mean gCF (av, ah, and, ap) or mean sCF (av, az, and cf). Concordance measurements for all topologies are available, see Methods and Data Availability.](figure_panels/Figure2_V4-01.png)

## Calibrating an expanded phylogeny to time {-}

Building on the phylotranscriptomic analyses above, we collected all publicly available genomic and transcriptomic data for species from Hawaiian _Drosophila_ and _Scaptomyza_. These data were accessioned in nine analyses published since 1997, most of which focused on resolving the phylogenetic relationships within a major clade[@baker1997multiple;@o2004phylogenetic;@bonacum2005phylogeny;@lapoint2011phylogenetic;@lapoint2013diversification;@lapoint2013diversification;@lapoint2014phylogenetics;@magnacca2015rapid;@katoh2017multiple]. The dataset we compiled contained `r length(gene_file_names)` genes (6 mitochondrial and `r length(gene_file_names)-6` nuclear) from `r ape::Ntip(root_all_tree)` species (including `r length(which(root_all_tree$tip.label %in% new_sp$name))` described and `r length(which(!root_all_tree$tip.label %in% new_sp$name))` undescribed putative species), with an overall occupancy of `r round(100*mean(sapply(new_fasta_list,length)/ape::Ntip(root_all_tree)),1)`\% (Fig. \ref{fig:new-fasta}). We used this dataset to infer the phylogeny with IQtree, constraining the relationships between major clades to conform to the topology shown in Figure 1A. 

The resulting topology is to our knowledge the most species rich phylogenetic tree of the Hawaiian Drosophilidae to date (Fig. \ref{fig:graphic-iqtree}). Several support values are low (ultrafast bootstrap <95), especially for nodes near the base of the radiation. However, this is not unexpected, given that this phylogeny is estimated primarily from the same data previously analyzed, which recovered alternative relationships at those nodes. Of note are the low support values for the relationships within the MM and _haleakalae_ clades (Fig. \ref{fig:graphic-iqtree}, polytomies), emphasizing the need for further study in these groups.

We used this expanded genetic dataset and topology to estimate the age of the Hawaiian Drosophilidae by calibrating this tree to time using the software package BEAST[@bouckaert2014beast]. Consistent with recent publications[@obbard2012estimating;@magnacca2015rapid;@katoh2017multiple], our results indicate that the age of the split between Hawaiian _Drosophila_ and _Scaptomyza_ occurred between 20 and 25 million years ago (Fig. 3, median root age `r round(beast_tree@data %>% arrange(desc(height)) %>% slice(1L) %>% pull(height_median),1)` million years). However, despite increased representation in both these groups, uncertainty around the root age remains substantial (95% highest posterior density confidence interval `r paste(round(as.numeric(unlist(stringr::str_split((beast_tree@data %>% arrange(desc(height)) %>% slice(1L))$height_0.95_HPD[[1]]," "))),1),collapse=" - ")` million years), and small changes in the calibration times used can lead to substantial differences in this estimate. The results shown here were calibrated using updated estimates for the ages at which Hawaiian islands became habitable, based on models of island emergence, growth, and decline via erosion and subsidance[@lim2017true]. However, calibrating with the same island age estimates as in Magnacca and Price (2015), which are marginally younger, we estimate the age of Hawaiian Drosophilidae to be ~15 million years old (median root age `r round(alt_beast_tree@data %>% arrange(desc(height)) %>% slice(1L) %>% pull(height_median),1)` million years). Furthermore, we note that calibrating using only vicariance based estimates of time is considered to be imprecise and should be avoided[@kodandaramaiah2011tectonic]. Taken together, we consider this estimate of the age of Hawaiian _Drosophila_, as well as those previously published, to be tentative, and suggest that further data (e.g. new fossil evidence) will be necessary to determine the age of diversification relative to island emergence with greater certainty.

According to this estimate, we find that the division between major Hawaiian _Drosophila_ clades occurred around ten million years ago (Fig 3), prior to the estimated time when the Hawaiian island Kaua'i became habitable (between 6.3 and 6.0 million years ago[@lim2017true]). Our results show that the diversification of lineages within MM also occured around that time, while the lineages within the AMC, _haleakalae_, and _grimshawi_ groups (PNA) all arose within the last five million years, around the time Oahu became habitable. We note that the MM groups suffers from lower representation across genes used to calibrate the tree to time (Fig. \ref{fig:new-fasta}), and suggest that more data may help shed light on differences in the age of this clade relative to others.

![**Time-calibrating the phylogeny of `r ape::Ntip(time_tree)` Drosophilidae species.** This phylogeny was inferred using IQtree to analyze all publically available genetic data for Hawaiian _Drosophila_ and _Scaptomyza_. It was then calibrated to time using the software BEAST, with four calibration points at nodes that show a biogeographic progression rule[@magnacca2015rapid]. The 95% highest posterior density intervals for each node are shown as gray bars, indicating the credible interval for the age of that group. The age at which four Hawaiian islands are estimated to have become habitable is shown in green. Colored labels indicate the clade to which taxa belong, and colors correspond to Fig 1; taxa without a colored label are species with genetic data that are as of yet undescribed. See Fig. \ref{fig:graphic-iqtree} for bootstrap support. Calibration using only island biogeography is known to be imprecise[@kodandaramaiah2011tectonic], therefore the divergence times shown here are considered tentative.](figure_panels/Figure3_V3-01.png)
 
## Ancestral state reconstruction of oviposition and larval feeding ecology {-}

With this time-calibrated tree for 316 species, we have an opportunity to investigate the evolutionary dynamics of trait diversification. By modeling the evolution of the diverse suite of ecological and morphological features across the phylogeny, we can identify which lineages have experienced major shifts in trait evolution. Predicting the number and phylogenetic position of these shifts will in turn be critical for informing future studies on development, life-history, and evolution of these flies. In the following analyses of trait evolution, we used the maximum clade credibility tree from the constrained BEAST analysis described above. Using this tree allows us to maximize the number of taxa for which genetic data are available, painting the most complete picture of ecological and morphological evolution in these flies up to this date. However, due to the fraction of genetic data missing across taxa, it also includes nodes with low bootstrap support (Fig. \ref{fig:graphic-iqtree}, polytomies). Therefore, for internal lineages for which evolutionary relationships remain uncertain, the position of these evolutionary shifts are subject to change as more genetic data becoem available and further phylogenetic resolution is achieved.

The Hawaiian Drosophilidae use a wide variety of plant, animal, and fungal species for egg laying and larval feeding (Fig. 4)[@heed1968ecology;@montgomery1975comparative;@magnacca2008review]. The majority of species breed in rotting substrates, with variation in the part of the plant or fungus in question, including rotting bark, leaves, flowers, and fruit. A few species breed on live tissue, and one notable _Scaptomyza_ subgenus, _Titanochaeta_, have been reared exclusively from spider egg masses[@knab1914titanochaeta]. In 2008, Magnacca and colleagues reviewed host plant and substrate records and found that, while many species can be considered specialists to species or substrate, host shifting was common and many species occasionally use non-preferred substrates[@magnacca2008review]. The type of oviposition substrate has been suggested as a driver for diversification of the reproductive traits ovariole number and egg size[@kambysellis1971studies;@kambysellis1995pattern;@sarikaya2019reproductive]. However, the previous reconstruction of oviposition substrate by Kambysellis and colleagues (1997)[@kambysellis1995pattern] was performed with a phylogeny that included only three non-PNA species, and was therefore unable to resolve the ancestral oviposition substrate for Hawaiian _Drosophila_ or to identify when evolutionary shifts in substrate outside of PNA were likely to have occurred.

We combined the phylogenetic results presented here with the data summarized in Magnacca and colleagues (2008), to reconstruct the ancestral oviposition substrate for the Hawaiian Drosophilidae (Fig. 5A, \ref{fig:reconstruct-eco-states}). Using stochastic character mapping[@huelsenbeck2003stochastic], we recover the most probable ancestral oviposition substrate for the Hawaiian _Drosophila_ as bark breeding (defined as including rearing records from bark, stems, branches, roots, and fern rachises, see Supplemental Table \ref{tab:substrate-categories}). We recover a transition from bark to leaf breeding at the base of the AMC clade that has generally persisted throughout the diversification of that group. As previously reported[@magnacca2008review], we find several groups that demonstrate no reported variation in substrate type (e.g. fungus breeding _haleakalae_, Fig. 5B).

Over 1000 stochastic character maps, we recovered an average of `r round(mean(pd$count[,1]),1)` transitions in oviposition substrate over the evolutionary history of Hawaiian Drosophilidae. The majority of these changes occurred along branches leading to extant tips, with few transitions at internal nodes (on the summary tree, `r nrow(int_node_changes)` out of `r nrow(node_changes)` total changes). On average, `r round(mean(apply(pd$count[,which(grepl("generalist",colnames(pd$count)))],1,sum)/apply(pd$count[,-1],1,sum)),1)*100`\% of transitions were between using a single substrate type as a primary host ("specialist" species) and using multiple types ("generalist" species, defined as using any two substrates that each comprise $>\frac{1}{4}$ of all rearing records, or with no substrate that comprises $>\frac{2}{3}$ of rearing records[@magnacca2008review].) Other transitions were primarily between using rotting bark, leaves, or sap. Pinpointing branches of likely transitions shows that some groups have experienced many more transitions than others, especially MM and non-flower / spider egg breeding _Scaptomyza_. Most generalist species fall in one of these two clades, which also include specialist bark and leaf breeders, among other substrates.

![**Ancestral state reconstruction of oviposition substrate indicates dozens of evolutionary transitions.** A, We used stochastic character mapping to reconstruct the ancestral substrate used for oviposition and larval feeding, and identified dozens of likely transitions in substrate (gray circles). Branch color indicates the ancestral substrate type with highest probability, and tip box indicates extant oviposition substrate. B, Oviposition substrate category was defined based on rearing records, using the data summarized in Magnacca and colleagues (2008)[@magnacca2008review]. Generalist species are defined as those with any two substrates that each comprise $>\frac{1}{4}$ of rearing records, or any species with no substrate that comprises $>\frac{2}{3}$ of rearing records.](figure_panels/Figure5_V4-01.png)

## Evolution of wing, thorax, and body length {-}

Alongside ecological diversification, the Hawaiian Drosophilidae show substantial diversity in adult body size. We used the time-calibrated phylogeny to model the number and timing of major changes in the evolutionary dynamics of size across the phylogeny. First, we digitized `r measurements %>% distinct(publication,species) %>% nrow()` records from 26 publications[@grimshaw1901fauna;@grimshaw1902fauna;@knab1914drosophilidae;@bryan1934review;@bryan1938key;@wirth1952two;@hackman1959genus;@hardy1965insects;@hardy1966viii;@hardy1968new;@hardy1969descriptions;@hardy1970notes;@hardy1971new;@hardy1972new;@hardy1977review;@hardy1979review;@kambysellis1971studies;@hardy1975studies;@o2001rustica;@hardy2001review;@o2003revision;@starmer2003phylogenetic;@magnacca2008revision;@magnacca2009revision;@craddock2016hawaiian;@sarikaya2019reproductive], including descriptions of body, wing, and thorax length across `r new_ms %>% distinct(name) %>% nrow()` species. Then we mapped these traits onto our phylogenetic results, and used the R package bayou[@uyeda2014novel] to identify branches that represent probable shifts in trait diversification. This package uses Ornstein-Uhlenbeck (OU) models to describe shifts in evolutionary regimes, defined as lineages that share an OU optimum trait value.

In the case of wing length, we find evidence for several highly supported regime shifts in the evolutionary history of Hawaiian Drosophilidae (Fig 5). Some of these are independent shifts on branches subtending groups with larger wings than their nearby relatives, including flies in the _antopocerus_ group (AMC) and in the _Engiscaptomyza_+_Grimshawomyia_ subgenera (_Scaptomyza_). Others are independent shifts on branches subtending lineages with smaller wings than nearby relatives such as the _nanella_+_ischnotrix_ (MM) and the _nudidrosophila_ subgroups (PNA). This suggests that the evolutionary history of Hawaiian _Drosophila_ has included multiple convergent transitions to both larger and smaller wings. In the case of _nudidrosophila_ (PNA), we note that the topology recovered in the summary tree dividing this subgroup into two lineages has very low bootstrap support (Fig. \ref{fig:graphic-iqtree}, polytomies), and we suggest that the two shifts to smaller wings recovered within PNA may represent a single shift if this group is indeed monophyletic.

We found similar results when considering thorax length (Fig. \ref{fig:print-thorax-bayou}) and body length (Fig: \ref{fig:print-body-bayou}). In the case of the former, we find shifts at the base of _antopocerus_, and _nudidrosophila_, consistent with the shifts recovered for wing size. In the case of body size, the most probable shifts are located at the base of the Hawaiian _Drosophila_ and the _Engiscaptomyza_+_Grimshawomyia_ subgenera. However for body length, no regime shifts received substantially more support than others, despite running bayou for an extra million generations and achieving a final effective size for log-likelihood of `r bayou_summary %>% filter(trait == "body length" & variable == "lnL") %>% pull("effective size")`.

## Evidence for convergent evolution of ovariole number and egg size {-}

We also performed these analyses on reproductive traits, including egg size, egg shape (aspect ratio, calculated as egg length / width), and the number of egg producing compartments in the ovary (ovarioles). These traits have been the subject of several life-history studies regarding the hypothesized trade-offs between offspring size and number, and its relationship to ecology[@kambysellis1971studies;@kambysellis1971studies;@starmer2003phylogenetic;@sarikaya2019reproductive]. Considering egg shape, we find evidence for a shift at the base of the PNA clade, which have proportionally longer eggs than their relatives (Fig. \ref{fig:reproductive-bayou}A). In the case of egg volume, we find evidence for independent shifts on branches subtending flies with large eggs (_antopocerus_ (AMC) and the _Engiscaptomyza_+_Grimshawomyia_, Fig. \ref{fig:reproductive-bayou}B). In the case of ovariole number, we find shifts at the base of the _haleakalae_+AMC+MM clades, which have on average fewer ovarioles than the other Hawaiian _Drosophila_ (_D. primaeva_ and PNA, Fig. \ref{fig:on-bayou}A).

Work by Kambyesllis and Heed (1971)[@kambysellis1971studies] suggested that Hawaiian Drosophilidae species can be grouped into four reproductive categories based on suites of ovarian and egg traits. Subsequent publications[@kambysellis1995pattern], including work by ourselves[@sarikaya2019reproductive], showed that these categories largely map to differences in oviposition substrate. Given the evidence that ovary and egg traits may be evolving together, we analyzed them with the R package SURFACE[@ingram2013surface], which uses OU models to analyze regime shifts in multiple traits at once, and allows for distant taxa to share regimes via convergent evolution. The best fitting model indicates four regimes (Fig: \ref{fig:reproductive-traits-surface}), two of which correspond to categories defined by Kambysellis and Heed (1971)[@kambysellis1971studies]: [1] very large eggs and low ovariole number in _S. undulata_ (_Grimshawomyia_) and _S. nasalis_ (_Engiscaptomyza_, group I in their publication); [2] large eggs with moderate to large bodies and moderate ovariole number in _antopocerus_ (AMC) and also in _S. crassifemur_ (_Engiscaptomyza_) and _S. ampliloba_ (_Engiscaptomyza_, group II in their publication). The remaining two regimes redistribute species that fall into groups IIIa and IIIb in Kambysellis and Heed (1971) into groups that have [3] small eggs, moderate to large bodies, and high ovariole number in PNA flies, _D. primaeva_ and _D. comatifemora_ (MM); [4] flies with small eggs, small to moderate bodies, and moderate ovariole number, in the remaining AMC+MM flies along with _D. preapicula_ (PNA) and _S. devexa_ (_Elmomyza_). As predicted by Kambysellis and collegues (1997)[@kambysellis1995pattern] and ourselves[@sarikaya2019reproductive], these final two regimes are largely divided between bark breeding flies (4) and leaf breeding flies (3). 

![**Multiple shifts in evolutionary regimes help explain the diversity of wing length.** A, Using the R package bayou[@uyeda2014novel], we modeled the evolution of wing length (mm) on the phylogeny and detected several probable shifts in evolutionary regimes (gray circles, larger indicates greater posterior probability that a shift occurred on that branch). Locations of probable shifts include at the base of the AMC+MM+_haleakalae_ clade, subtending the _antopocerus_ group (AMC), and subtending the _nudidrosophila_ (PNA), among others. B, The distribution of wing lengths across the phylogeny of Hawaiian _Drosophila_ and _Scaptomyza_.](figure_panels/Figure4_V1-01.png)

# Discussion {-}

The landscape of treespace, representing support for all the possible topologies given the data, is often hidden from our view[@sanderson2011terraces;@st2017shape]. This is especially true as the size of datasets grow, making it more laborious to traverse treespace landscapes. Approaches such as visualizing the posterior distribution of parameters in a Bayesian analysis, or alternative hypotheses testing (e.g. an SOWH test in a maximum likelihood framework), can provide a sense of how support for one result compares to others. But given that a complete exploration of treespace is typically not available, we often do not know whether the support landscape in treespace is generally flat, rugged, or highly structured.

Model clades for phylogenetics such as the Hawaiian Drosophilidae, however, offer an opportunity to explore these methods using real-world data. In the case of the landscape of treespace, especially in the context of discordance of gene trees and species trees, these flies have a long history as one such model clade. Here we provide a comprehensive snapshot of treespace for this island radiation. We find that, in this case, the landscape of support is largely defined by one hotspot in both gene and site concordance. This hotspot divides the major clades of Hawaiian _Drosophila_ into two main lineages, the picture wing flies and their allies (PNA) on one side, and the _modified-mouthparts_ and _modified-tarsus_ (AMC) flies on the other. We consider this division to be strongly indicated given the data, and we note that this is in line with other recent phylogenetic results (Fig. \ref{fig:previous-phylogenetic-hypotheses})[@o2011phylogenetic;@magnacca2015rapid].

Within this hotspot of support, several alternative topologies that differ in the placement of smaller clades (_D. primaeva_ and _haleakalae_) have an equivocal amount of support across genes and sites. We suggest that much of this discordance represents the results of evolutionary processes that took place on the short internodes at the base of the radiation. Despite this local discordance, the outcome of all phylogenetic software tested here indicates strong support for a single topology (Fig. 1A). With this information, we consider that tree, with PNA as the sister clade to the rest of Hawaiian _Drosophila_, and _haleakalae_ as the sister clade to AMC+MM, to be a plausible new hypothesis for the evolution of these flies. We suggest that additional taxonomic sampling in the _haleakalae_ will be valuable in gaining a fine-scale view of the landscape of support within this hotspot.

This new hypothesis for the relationship between major groups has several implications for our understanding of ecological and morphological evolution. Some previous studies have focused on defining one group as 'basal' to others (e.g. _haleakalae_, MM, or _D. primaeva_)[@kambysellis1995pattern;@o2011phylogenetic]. However our results provide an alternative interpretation. We find that the PNA clade (including _picture-wing_ flies) is the sister clade to all others, and we note that for at least one trait (bark breeding), most PNA flies appear to display the same state as the most common ancestor of Hawaiian _Drosophila_. The relationship between this group to _haleakalae_ and others suggests the possibility of a secondary loss of complex courtship behavior in the latter[@o2011phylogenetic]. We note that the overall pattern in the group has been one of many transitions to and from the ancestral state, including in ecology, size, and allometry.

Our results on wing, body, and egg size evolution show that Hawaiian Drosophilidae have experienced multiple, independent shifts to both larger and smaller sizes. These repeated changes present an opportunity to test the predictability of evolution by analyzing whether repeated changes in size are coincident with changes in other features, including ecology, development, and whether these repeated trait changes share the same genetic regulatory basis. The findings of this study on ovariole number and egg size evolution are consistent with what has previously been shown[@kambysellis1995pattern;@sarikaya2019reproductive], indicating that evolutionary changes in the larval ecology correspond to changes in reproductive trait evolution. However, our findings here show that larval feeding substrate does not explain all the dynamics of trait diversification in Hawaiian _Drosophila_. For example, the _antopocerus_ group (AMC) shares the same oviposition substrate as most other AMC flies, yet we find evidence that several important shifts in thorax, wing, and egg size evolution all occurred on the branch subtending its diversification.

Preivous authors have commented on the potential of the Hawaiian _Drosophila_ as a model clade for the study of the evolution of development[@edwards2007database;@o2018hawaiian], given its close relationship to genetic model species like _D. melanogaster_. Progress in this effort has not always been straightforward, however, given their longer generation times and specific host plant requirements to induce oviposition in the lab[@o2018hawaiian]. We propose that advances in evo-devo study of the Hawaiian Drosophilidae will be added by leveraging evolutionary methods to formulate and test developmental hypotheses. For example, we can use phylogenetic comparative methods to statistically detect signatures of convergent evolution and to identify changes in patterns of allometric growth[@stevenson1995organ;@sarikaya2019reproductive]. Going forward, such methods will be essential in providing testable hypotheses regarding the relationship of developmental data to ecological and morphological parameters. The results of these analyses will provide valuable complementary studies to the developmental literature generated using laboratory-amenable model drosophilids, and shed light on the genetic basis of this remarkable island radiation.

# Acknowledgments {-}

This work was partially supported by National Science Foundation Graduate Research Fellowship Program DGE1745303 to SHC,  National Institutes of Health award R01 HD073499-01 (NICHD) to CGE, and funds from Harvard University to support SHC and CGE. We thank Didem Sarikaya, Karl Magnacca, and Steve Montgomery for their field assistance and expertise in Hawaiian fly identification and husbandry. We thank Kenneth Kaneshiro for the use of his lab space in preparing wild caught specimens. We thank Tauana Cunha and Bruno de Medeiros for providing scripts to run phylogenetic software on Harvard's computing cluster, and Casey Dunn for providing scripts to handle large genetic data files. We thank members of the Extavour Lab for discussion of ideas.

# References {-}



