<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<title>Drosophila macrothrix</title>
<link href="css/bootstrap.min.css" rel="stylesheet" media="screen">
<style type="text/css">
body {
  margin: 30px 50px;
}
.stats {
  margin: 10px;
  display: none;
}
.table {
  font-size: 14px;
  font-family: Monaco,Menlo,Consolas,"Courier New",monospace;
}
.table-mini {
  width: 50%;
}
div.stage { padding: 10px 0 10px 0; border-top: 5px solid #eee; }
blockquote p { font-size: 14px; }
.btn-lg,
.btn-group-lg > .btn {
  padding: 10px 16px;
  font-size: 18px;
  line-height: 1.33;
  border-radius: 6px;
}
</style>
<script type="text/javascript">
function togglestats(element){
element.nextElementSibling.style.display == "none" || element.nextElementSibling.style.display == "" ?
element.nextElementSibling.style.display = "table" :
element.nextElementSibling.style.display = "none";
}
</script>

</style>
</head>
<body>
<h1><em>Drosophila macrothrix</em></h1>
<blockquote>
<table class="table table-striped table-condensed">
<tr><td><b>id</b></td><td>055A</td></tr>
<tr><td><b>ncbi_id</b></td><td>46811</td></tr>
<tr><td><b>itis_id</b></td><td>146212</td></tr>
</table>
</blockquote>
<h2 id="toc">Table of Contents</a></h2><table class="table table-striped">

<tr class="minimal">
<td class="minimal">
<b>Run 7</b><br/><i>qc</i><br/>2021-06-09T16:48:32.334939<br/>holy7c04401.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#qc_7">qc</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :       625.20
User Time (s)      :       563.10
System Time (s)    :         8.28
Max Memory (KB)    :    1,607,268</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 20</b><br/><i>transcriptome</i><br/>2021-06-13T00:00:59.231859<br/>holy7c04401.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#insert_size_20">insert_size</a></li><li><a href="#rrna_20">rrna</a></li><li><a href="#assemble_20">assemble</a></li><li><a href="#translate_20">translate</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :     22743.21
User Time (s)      :    262941.60
System Time (s)    :     18726.59
Max Memory (KB)    :   67,694,468</pre>
</td>
</tr>
</table>
<div id="qc_7" class="stage">
<h2>qc <small>(Run 7)</small></h2>
<blockquote>
<p>Quality control for raw Illumina reads using FastQC.</p>

</blockquote>
<h4>FastQC Summary</h4>
<table class="table">
<tr><th></th>
<th>FASTQ 1 (<a href="fastqc1.html">full report</a>)</th>
<th>FASTQ 2 (<a href="fastqc2.html">full report</a>)</th>
</tr>
<tr><th>Basic Statistics</th>
<td><span style="color:green;">PASS</span></td>
<td><span style="color:green;">PASS</span></td>
</tr>
<tr><th>Per base sequence quality</th>
<td><span style="color:green;">PASS</span></td>
<td><span style="color:green;">PASS</span></td>
</tr>
<tr><th>Per tile sequence quality</th>
<td><span style="color:orange;">WARN</span></td>
<td><span style="color:green;">PASS</span></td>
</tr>
<tr><th>Per sequence quality scores</th>
<td><span style="color:green;">PASS</span></td>
<td><span style="color:green;">PASS</span></td>
</tr>
<tr><th>Per base sequence content</th>
<td><span style="color:red;">FAIL</span></td>
<td><span style="color:red;">FAIL</span></td>
</tr>
<tr><th>Per sequence GC content</th>
<td><span style="color:orange;">WARN</span></td>
<td><span style="color:red;">FAIL</span></td>
</tr>
<tr><th>Per base N content</th>
<td><span style="color:green;">PASS</span></td>
<td><span style="color:green;">PASS</span></td>
</tr>
<tr><th>Sequence Length Distribution</th>
<td><span style="color:orange;">WARN</span></td>
<td><span style="color:orange;">WARN</span></td>
</tr>
<tr><th>Sequence Duplication Levels</th>
<td><span style="color:red;">FAIL</span></td>
<td><span style="color:red;">FAIL</span></td>
</tr>
<tr><th>Overrepresented sequences</th>
<td><span style="color:red;">FAIL</span></td>
<td><span style="color:red;">FAIL</span></td>
</tr>
<tr><th>Adapter Content</th>
<td><span style="color:red;">FAIL</span></td>
<td><span style="color:red;">FAIL</span></td>
</tr>
<tr><th>Kmer Content</th>
<td><span style="color:red;">FAIL</span></td>
<td><span style="color:red;">FAIL</span></td>
</tr>
</table>
<blockquote><p><a class="reference external" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> is a tool from Babraham Bioinformatics that generates detailed
quality diagnostics of NGS sequence data.</p>
</blockquote>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">296.33 [sum]</td>
<td class="right">282.38 [sum]</td>
<td class="right">4.01 [sum]</td>
<td class="right">1,542,244 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
fastqc
</td>
<td>
fastqc
</td>
<td class="right">
296.33
</td>
<td class="right">
282.38
</td>
<td class="right">
4.01
</td>
<td class="right">
1,542,244
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="insert_size_20" class="stage">
<h2>insert_size <small>(Run 20)</small></h2>
<blockquote>
<p>Estimates the insert size distribution of paired-end Illumina data by
assembling a subset of the data and mapping read pairs to it. The insert
size does not include the adapters added during library preparation.</p>

</blockquote>
<table class="table table-striped table-condensed table-mini">
<tr><td><b>Mean insert size (bp)</b></td><td class="right">185.05</td></tr>
<tr><td><b>Standard deviation (bp)</b></td><td class="right">51.34</td></tr>
</table>
<img src="20.insert.hist.png"/>
<blockquote><p>A histogram of insert sizes.</p>
</blockquote>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">281.42 [sum]</td>
<td class="right">2284.47 [sum]</td>
<td class="right">652.35 [sum]</td>
<td class="right">1,760,164 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
filter_illumina
</td>
<td>
assemble_subset
</td>
<td class="right">
3.16
</td>
<td class="right">
1.66
</td>
<td class="right">
0.64
</td>
<td class="right">
64,104
</td>
</tr>
<tr>
<td>
trinity
</td>
<td>
assemble_subset
</td>
<td class="right">
272.33
</td>
<td class="right">
2207.76
</td>
<td class="right">
649.56
</td>
<td class="right">
1,760,164
</td>
</tr>
<tr>
<td>
bowtie2-build
</td>
<td>
estimate_insert
</td>
<td class="right">
0.84
</td>
<td class="right">
0.54
</td>
<td class="right">
0.07
</td>
<td class="right">
98,856
</td>
</tr>
<tr>
<td>
bowtie2
</td>
<td>
estimate_insert
</td>
<td class="right">
5.08
</td>
<td class="right">
74.50
</td>
<td class="right">
2.09
</td>
<td class="right">
937,204
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="rrna_20" class="stage">
<h2>rrna <small>(Run 20)</small></h2>
<blockquote>
<p>Assembles and identifies ribosomal RNA (rRNA) sequences, removes read pairs
that map to these rRNA sequences, and provides a variety of diagnostics about
rRNA. A single exemplar sequence is presented for each type of rRNA that is
found, but rRNA read pairs are excluded by mapping to a large set of rRNA
transcripts that are derived from multiple assemblies over a range of data
subset sizes.</p>

</blockquote>
<table class="table table-striped table-condensed table-mini">
<tr><td><b>Read pairs examined</b></td><td class="right">40,293,896</td></tr>
<tr><td><b>Read pairs kept</b></td><td class="right">27,394,368</td></tr>
<tr><td><b>Percent kept</b></td><td class="right">68.0%</td></tr>
</table>
<table class="table table-striped table-condensed table-mini">
<tr>
<td>Gene</td>
<td style="text-align:right">Mapped Reads</td>
<td style="text-align:right">Percent</td>
</tr>
<tr>
<td>large-nuclear-rRNA</td>
<td style="text-align:right">22,391,924</td>
<td style="text-align:right">27.8%</td>
</tr>
</table>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">914.56 [sum]</td>
<td class="right">4834.12 [sum]</td>
<td class="right">1237.79 [sum]</td>
<td class="right">1,957,864 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
filter_illumina
</td>
<td>
assemble_subsets.1000
</td>
<td class="right">
0.10
</td>
<td class="right">
0.02
</td>
<td class="right">
0.05
</td>
<td class="right">
65,200
</td>
</tr>
<tr>
<td>
trinity
</td>
<td>
assemble_subsets.1000
</td>
<td class="right">
10.79
</td>
<td class="right">
28.58
</td>
<td class="right">
4.50
</td>
<td class="right">
735,660
</td>
</tr>
<tr>
<td>
filter_illumina
</td>
<td>
assemble_subsets.2500
</td>
<td class="right">
0.08
</td>
<td class="right">
0.03
</td>
<td class="right">
0.04
</td>
<td class="right">
65,228
</td>
</tr>
<tr>
<td>
trinity
</td>
<td>
assemble_subsets.2500
</td>
<td class="right">
14.01
</td>
<td class="right">
53.74
</td>
<td class="right">
7.84
</td>
<td class="right">
737,660
</td>
</tr>
<tr>
<td>
filter_illumina
</td>
<td>
assemble_subsets.5000
</td>
<td class="right">
0.12
</td>
<td class="right">
0.06
</td>
<td class="right">
0.04
</td>
<td class="right">
65,588
</td>
</tr>
<tr>
<td>
trinity
</td>
<td>
assemble_subsets.5000
</td>
<td class="right">
14.11
</td>
<td class="right">
75.87
</td>
<td class="right">
11.53
</td>
<td class="right">
748,040
</td>
</tr>
<tr>
<td>
filter_illumina
</td>
<td>
assemble_subsets.10000
</td>
<td class="right">
0.18
</td>
<td class="right">
0.08
</td>
<td class="right">
0.08
</td>
<td class="right">
65,608
</td>
</tr>
<tr>
<td>
trinity
</td>
<td>
assemble_subsets.10000
</td>
<td class="right">
19.54
</td>
<td class="right">
137.81
</td>
<td class="right">
18.46
</td>
<td class="right">
777,008
</td>
</tr>
<tr>
<td>
filter_illumina
</td>
<td>
assemble_subsets.25000
</td>
<td class="right">
0.39
</td>
<td class="right">
0.20
</td>
<td class="right">
0.16
</td>
<td class="right">
65,620
</td>
</tr>
<tr>
<td>
trinity
</td>
<td>
assemble_subsets.25000
</td>
<td class="right">
39.03
</td>
<td class="right">
318.78
</td>
<td class="right">
38.33
</td>
<td class="right">
892,812
</td>
</tr>
<tr>
<td>
filter_illumina
</td>
<td>
assemble_subsets.50000
</td>
<td class="right">
0.71
</td>
<td class="right">
0.39
</td>
<td class="right">
0.30
</td>
<td class="right">
65,628
</td>
</tr>
<tr>
<td>
trinity
</td>
<td>
assemble_subsets.50000
</td>
<td class="right">
75.91
</td>
<td class="right">
690.49
</td>
<td class="right">
85.95
</td>
<td class="right">
1,550,300
</td>
</tr>
<tr>
<td>
filter_illumina
</td>
<td>
assemble_subsets.100000
</td>
<td class="right">
1.39
</td>
<td class="right">
0.78
</td>
<td class="right">
0.58
</td>
<td class="right">
65,628
</td>
</tr>
<tr>
<td>
trinity
</td>
<td>
assemble_subsets.100000
</td>
<td class="right">
132.02
</td>
<td class="right">
1241.80
</td>
<td class="right">
150.74
</td>
<td class="right">
1,613,868
</td>
</tr>
<tr>
<td>
makeblastdb
</td>
<td>
blast_transcripts
</td>
<td class="right">
0.07
</td>
<td class="right">
0.02
</td>
<td class="right">
0.02
</td>
<td class="right">
65,644
</td>
</tr>
<tr>
<td>
blastn
</td>
<td>
blast_transcripts
</td>
<td class="right">
0.43
</td>
<td class="right">
0.22
</td>
<td class="right">
0.14
</td>
<td class="right">
116,792
</td>
</tr>
<tr>
<td>
bowtie2-build
</td>
<td>
map_reads
</td>
<td class="right">
0.44
</td>
<td class="right">
0.13
</td>
<td class="right">
0.08
</td>
<td class="right">
98,176
</td>
</tr>
<tr>
<td>
bowtie2
</td>
<td>
map_reads
</td>
<td class="right">
182.64
</td>
<td class="right">
2028.87
</td>
<td class="right">
753.69
</td>
<td class="right">
569,136
</td>
</tr>
<tr>
<td>
exclude
</td>
<td>
exclude_reads
</td>
<td class="right">
422.59
</td>
<td class="right">
256.26
</td>
<td class="right">
165.27
</td>
<td class="right">
1,957,864
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="assemble_20" class="stage">
<h2>assemble <small>(Run 20)</small></h2>
<blockquote>
<p>Assembles reads into transcripts, processes the assembly, and generates
assembly diagnostics. Read pairs are first filtered at a more stringent
mean quality threshold. Transcriptome assembly is generated with Trinity
assembler.</p>

</blockquote>
<h4>Illumina Filtering</h4>
<table class="table table-striped table-condensed table-mini">
<tr><td><b>Read pairs examined</b></td><td class="right">27,394,368</td></tr>
<tr><td><b>Read pairs kept</b></td><td class="right">21,877,845</td></tr>
<tr><td><b>Percent kept</b></td><td class="right">79.9%</td></tr>
<tr><td><b>Illumina quality threshold</b></td><td class="right">33</td></tr>
<tr><td><b>Adapter fails</b></td><td class="right">3,762,598</td></tr>
<tr><td><b>Quality fails</b></td><td class="right">1,979,130</td></tr>
<tr><td><b>Base composition fails</b></td><td class="right">386,230</td></tr>
</table>
<h4>Sequence Types</h4>
<table class="table table-striped table-condensed table-mini">
<tr><td></td>
<td><b>protein-coding</b></td>
<td><b>ribosomal-large</b></td>
<td><b>ribosomal-small</b></td>
<td><b>unknown</b></td>
</tr>
</tr><td><b>mitochondrial</b></td>
<td>18</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
</tr><td><b>nuclear</b></td>
<td>35,956</td>
<td>49</td>
<td>58</td>
<td>0</td>
</tr>
</tr><td><b>plastid</b></td>
<td>3</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
</tr><td><b>unknown</b></td>
<td>16,746</td>
<td>0</td>
<td>0</td>
<td>54,356</td>
</tr>
</tr><td><b>vector</b></td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>44</td>
</tr>
</table>
<blockquote><p>Table showing classification of genome and molecule types for the
sequences.</p>
</blockquote>
<img src="20.assembly.coverage.svg"/>
<h4>Confidence vs. Expression</h4>
<img src="20.assembly.confidence.png" />
<div class="well well-sm" style="text-align:center">
<h4>Downloads</h4>
<a href="20.assembly.fa" class="btn btn-lg btn-success">Assembly FASTA</a>
</div>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">16422.96 [sum]</td>
<td class="right">181807.47 [sum]</td>
<td class="right">15824.00 [sum]</td>
<td class="right">67,120,940 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
filter_illumina
</td>
<td>
filter_data
</td>
<td class="right">
610.07
</td>
<td class="right">
481.12
</td>
<td class="right">
127.49
</td>
<td class="right">
69,260
</td>
</tr>
<tr>
<td>
trinity
</td>
<td>
assemble
</td>
<td class="right">
12419.53
</td>
<td class="right">
148700.29
</td>
<td class="right">
14065.42
</td>
<td class="right">
67,120,940
</td>
</tr>
<tr>
<td>
parallel
</td>
<td>
remove_vectors
</td>
<td class="right">
5.82
</td>
<td class="right">
34.17
</td>
<td class="right">
12.88
</td>
<td class="right">
420,628
</td>
</tr>
<tr>
<td>
makeblastdb
</td>
<td>
remove_rrna
</td>
<td class="right">
0.07
</td>
<td class="right">
0.02
</td>
<td class="right">
0.02
</td>
<td class="right">
420,624
</td>
</tr>
<tr>
<td>
parallel
</td>
<td>
remove_rrna
</td>
<td class="right">
3.68
</td>
<td class="right">
28.52
</td>
<td class="right">
12.26
</td>
<td class="right">
420,624
</td>
</tr>
<tr>
<td>
rsem-eval-calculate-score
</td>
<td>
estimate_confidence
</td>
<td class="right">
3383.79
</td>
<td class="right">
32563.35
</td>
<td class="right">
1605.93
</td>
<td class="right">
2,298,364
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="translate_20" class="stage">
<h2>translate <small>(Run 20)</small></h2>
<blockquote>
<p>Cleans transcripts to remove ribosomal, mitochondrial, vector, and
low-complexity sequences.  Vector sequences could include untrimmed adapters or
plasmids (we sometimes find sequences in our data for the protein expression
vectors used to manufacture the sample preparation enzymes).  Raw reads are
mapped back to the transcripts to estimate coverage and assign TPKM values.
Finally, transcripts are annotated with blast hits against SwissProt.</p>

</blockquote>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">4782.77 [sum]</td>
<td class="right">72023.24 [sum]</td>
<td class="right">74.98 [sum]</td>
<td class="right">425,668 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
TransDecoder.LongOrfs
</td>
<td>
identify_orfs.TransDecoder
</td>
<td class="right">
108.00
</td>
<td class="right">
106.46
</td>
<td class="right">
0.69
</td>
<td class="right">
425,668
</td>
</tr>
<tr>
<td>
parallel
</td>
<td>
annotate_orfs
</td>
<td class="right">
4674.78
</td>
<td class="right">
71916.78
</td>
<td class="right">
74.29
</td>
<td class="right">
425,668
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
</body></html>