<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<title>assembly_phylogeny_expression_Hawaiian_Drosophila_June2021</title>
<link href="css/bootstrap.min.css" rel="stylesheet" media="screen">
<style type="text/css">
body {
  margin: 30px 50px;
}
.stats {
  margin: 10px;
  display: none;
}
.table {
  font-size: 14px;
  font-family: Monaco,Menlo,Consolas,"Courier New",monospace;
}
.table-mini {
  width: 50%;
}
div.stage { padding: 10px 0 10px 0; border-top: 5px solid #eee; }
blockquote p { font-size: 14px; }
.btn-lg,
.btn-group-lg > .btn {
  padding: 10px 16px;
  font-size: 18px;
  line-height: 1.33;
  border-radius: 6px;
}
</style>
<script type="text/javascript">
function togglestats(element){
element.nextElementSibling.style.display == "none" || element.nextElementSibling.style.display == "" ?
element.nextElementSibling.style.display = "table" :
element.nextElementSibling.style.display = "none";
}
</script>

</style>
</head>
<body>
<h1><em>assembly_phylogeny_expression_Hawaiian_Drosophila_June2021</em></h1>
<blockquote>
<table class="table table-striped table-condensed">
<tr><td><b>id</b></td><td>assembly_phylogeny_expression_Hawaiian_Drosophila_June2021</td></tr>
</table>
</blockquote>
<h2 id="toc">Table of Contents</a></h2><table class="table table-striped">

<tr class="minimal">
<td class="minimal">
<b>Run 50</b><br/><i>homologize</i><br/>2021-06-15T01:34:48.030796<br/>holy7c18609.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#homologize_50">homologize</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :     17099.46
User Time (s)      :    400590.01
System Time (s)    :       470.74
Max Memory (KB)    :      954,576</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 51</b><br/><i>multalign</i><br/>2021-06-15T06:19:55.671542<br/>holy7c18609.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#multalign_51">multalign</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :    130911.35
User Time (s)      :    785239.28
System Time (s)    :    104235.70
Max Memory (KB)    :  155,185,512</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 52</b><br/><i>genetree</i><br/>2021-06-16T18:42:10.897741<br/>holy7c18609.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#genetree_52">genetree</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :      1440.54
User Time (s)      :     33116.33
System Time (s)    :        70.05
Max Memory (KB)    :      474,676</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 53</b><br/><i>treeinform</i><br/>2021-06-16T19:06:27.315845<br/>holy7c18609.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :        55.66
User Time (s)      :        28.40
System Time (s)    :         1.17
Max Memory (KB)    :      675,184</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 54</b><br/><i>homologize</i><br/>2021-06-16T19:26:56.029560<br/>holy7c18609.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#homologize_54">homologize</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :     16827.28
User Time (s)      :    394961.58
System Time (s)    :       475.33
Max Memory (KB)    :      949,840</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 55</b><br/><i>multalign</i><br/>2021-06-17T00:07:32.028903<br/>holy7c18609.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#multalign_55">multalign</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :    130547.18
User Time (s)      :    780635.62
System Time (s)    :    103616.90
Max Memory (KB)    :  155,193,412</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 56</b><br/><i>genetree</i><br/>2021-06-18T12:23:27.777957<br/>holy7c18609.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#genetree_56">genetree</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :      1385.47
User Time (s)      :     32359.69
System Time (s)    :        90.52
Max Memory (KB)    :      471,452</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 57</b><br/><i>treeprune</i><br/>2021-06-18T12:46:43.222883<br/>holy7c18609.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#treeprune_57">treeprune</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :       198.93
User Time (s)      :       194.57
System Time (s)    :         4.28
Max Memory (KB)    :    6,009,608</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 58</b><br/><i>multalign</i><br/>2021-06-19T00:37:00.228364<br/>holy7c18608.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#multalign_58">multalign</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :     47802.12
User Time (s)      :    131067.44
System Time (s)    :     65039.73
Max Memory (KB)    :   53,601,988</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 59</b><br/><i>supermatrix</i><br/>2021-06-19T13:53:57.410041<br/>holy7c18608.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#supermatrix_59">supermatrix</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :        24.87
User Time (s)      :        21.30
System Time (s)    :         3.31
Max Memory (KB)    :    2,048,832</pre>
</td>
</tr>

<tr class="minimal">
<td class="minimal">
<b>Run 60</b><br/><i>supermatrix</i><br/>2021-06-19T20:13:29.171250<br/>holy7c26601.rc.fas.harvard.edu
</td>
<td class="minimal">
<ul><li><a href="#supermatrix_60">supermatrix</a></li></ul>
</td>
<td class="minimal">
<pre>Wall Time (s)      :        26.07
User Time (s)      :        21.07
System Time (s)    :         2.58
Max Memory (KB)    :      927,252</pre>
</td>
</tr>
</table>
<div id="homologize_50" class="stage">
<h2>homologize <small>(Run 50)</small></h2>
<blockquote>
<p>Identifies homologous sequences across datasets. Takes assembly or group of
assemblies and prepares a set of comprehensive comparisons between them. First
an all by all BLAST is run with a stringent threshold, and the hits which match
above a given score are used as edges between two transcripts which then form a
graph. This graph is the basis of a series of comparative scores.</p>

</blockquote>
<p>Total sequences: 217918</p>
<blockquote><p>Summary of all species processed.</p>
</blockquote>
<img src="50.cluster.hist.png"/>
<blockquote><p>Distribution of the number of nodes in each cluster.</p>
</blockquote>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">17080.43 [sum]</td>
<td class="right">400544.65 [sum]</td>
<td class="right">468.27 [sum]</td>
<td class="right">472,692 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
makeblastdb
</td>
<td>
prepare_blast
</td>
<td class="right">
62.61
</td>
<td class="right">
17.73
</td>
<td class="right">
15.15
</td>
<td class="right">
395,916
</td>
</tr>
<tr>
<td>
parallel
</td>
<td>
run_blast
</td>
<td class="right">
16962.73
</td>
<td class="right">
400474.25
</td>
<td class="right">
235.03
</td>
<td class="right">
396,160
</td>
</tr>
<tr>
<td>
mcxload
</td>
<td>
parse_edges
</td>
<td class="right">
35.08
</td>
<td class="right">
10.55
</td>
<td class="right">
0.42
</td>
<td class="right">
396,340
</td>
</tr>
<tr>
<td>
mcl
</td>
<td>
mcl_cluster
</td>
<td class="right">
20.01
</td>
<td class="right">
42.13
</td>
<td class="right">
217.67
</td>
<td class="right">
472,692
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="multalign_51" class="stage">
<h2>multalign <small>(Run 51)</small></h2>
<blockquote>
<p>Applies sampling and length filters to each cluster of homologous sequences.</p>
<p>Creates multiple sequence alignments for each cluster of homologous sequences
using MAFFT using the E-INS-i algorithm; see doi: 10.1093/molbev/mst010.</p>
<p>Cleans up alignments using Gblocks.</p>

</blockquote>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">130802.21 [sum]</td>
<td class="right">785190.21 [sum]</td>
<td class="right">104228.53 [sum]</td>
<td class="right">154,542,264 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
parallel
</td>
<td>
align_sequences
</td>
<td class="right">
130765.07
</td>
<td class="right">
785016.93
</td>
<td class="right">
104170.81
</td>
<td class="right">
154,542,264
</td>
</tr>
<tr>
<td>
parallel
</td>
<td>
cleanup_alignments
</td>
<td class="right">
37.14
</td>
<td class="right">
173.28
</td>
<td class="right">
57.71
</td>
<td class="right">
636,172
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="genetree_52" class="stage">
<h2>genetree <small>(Run 52)</small></h2>
<blockquote>
<p>Builds gene trees for each alignment of homologous sequences, it builds a
phylogenetic tree using the maximum likelihood optimality criterion as
implemented in RAxML (see <a class="reference external" href="http://www.exelixis-lab.org/">http://www.exelixis-lab.org/</a> and
doi:10.1093/bioinformatics/btl446).</p>
<p>Use --previous for a specific set of alignments, otherwise this pipeline will
search for the output from the most recent run of multalign for the given
catalog ID.</p>

</blockquote>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">1401.23 [sum]</td>
<td class="right">33112.19 [sum]</td>
<td class="right">65.72 [sum]</td>
<td class="right">226,768 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
parallel
</td>
<td>
genetrees
</td>
<td class="right">
1401.23
</td>
<td class="right">
33112.19
</td>
<td class="right">
65.72
</td>
<td class="right">
226,768
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="homologize_54" class="stage">
<h2>homologize <small>(Run 54)</small></h2>
<blockquote>
<p>Identifies homologous sequences across datasets. Takes assembly or group of
assemblies and prepares a set of comprehensive comparisons between them. First
an all by all BLAST is run with a stringent threshold, and the hits which match
above a given score are used as edges between two transcripts which then form a
graph. This graph is the basis of a series of comparative scores.</p>

</blockquote>
<p>Total sequences: 215090</p>
<blockquote><p>Summary of all species processed.</p>
</blockquote>
<img src="54.cluster.hist.png"/>
<blockquote><p>Distribution of the number of nodes in each cluster.</p>
</blockquote>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">16786.07 [sum]</td>
<td class="right">394915.20 [sum]</td>
<td class="right">472.93 [sum]</td>
<td class="right">467,956 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
makeblastdb
</td>
<td>
prepare_blast
</td>
<td class="right">
58.68
</td>
<td class="right">
15.74
</td>
<td class="right">
13.76
</td>
<td class="right">
395,316
</td>
</tr>
<tr>
<td>
parallel
</td>
<td>
run_blast
</td>
<td class="right">
16673.96
</td>
<td class="right">
394847.98
</td>
<td class="right">
241.72
</td>
<td class="right">
395,564
</td>
</tr>
<tr>
<td>
mcxload
</td>
<td>
parse_edges
</td>
<td class="right">
34.24
</td>
<td class="right">
10.29
</td>
<td class="right">
0.40
</td>
<td class="right">
395,744
</td>
</tr>
<tr>
<td>
mcl
</td>
<td>
mcl_cluster
</td>
<td class="right">
19.20
</td>
<td class="right">
41.19
</td>
<td class="right">
217.06
</td>
<td class="right">
467,956
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="multalign_55" class="stage">
<h2>multalign <small>(Run 55)</small></h2>
<blockquote>
<p>Applies sampling and length filters to each cluster of homologous sequences.</p>
<p>Creates multiple sequence alignments for each cluster of homologous sequences
using MAFFT using the E-INS-i algorithm; see doi: 10.1093/molbev/mst010.</p>
<p>Cleans up alignments using Gblocks.</p>

</blockquote>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">130368.67 [sum]</td>
<td class="right">780583.00 [sum]</td>
<td class="right">103607.64 [sum]</td>
<td class="right">154,547,052 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
parallel
</td>
<td>
align_sequences
</td>
<td class="right">
130334.75
</td>
<td class="right">
780415.79
</td>
<td class="right">
103551.47
</td>
<td class="right">
154,547,052
</td>
</tr>
<tr>
<td>
parallel
</td>
<td>
cleanup_alignments
</td>
<td class="right">
33.92
</td>
<td class="right">
167.20
</td>
<td class="right">
56.17
</td>
<td class="right">
631,616
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="genetree_56" class="stage">
<h2>genetree <small>(Run 56)</small></h2>
<blockquote>
<p>Builds gene trees for each alignment of homologous sequences, it builds a
phylogenetic tree using the maximum likelihood optimality criterion as
implemented in RAxML (see <a class="reference external" href="http://www.exelixis-lab.org/">http://www.exelixis-lab.org/</a> and
doi:10.1093/bioinformatics/btl446).</p>
<p>Use --previous for a specific set of alignments, otherwise this pipeline will
search for the output from the most recent run of multalign for the given
catalog ID.</p>

</blockquote>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">1371.36 [sum]</td>
<td class="right">32356.59 [sum]</td>
<td class="right">86.75 [sum]</td>
<td class="right">225,108 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
parallel
</td>
<td>
genetrees
</td>
<td class="right">
1371.36
</td>
<td class="right">
32356.59
</td>
<td class="right">
86.75
</td>
<td class="right">
225,108
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="treeprune_57" class="stage">
<h2>treeprune <small>(Run 57)</small></h2>
<blockquote>
<p>For each gene tree generated in genetree, prune the tree to include only one
representative sequence per taxon when sequences form a monophyletic group
(here called 'monophyly masking'). Then prune the monophyly-masked tree into
maximally inclusive subtrees with no more than one sequence per taxon (here
called 'paralogy pruning').</p>
<p>Use --previous for a specific set of trees, otherwise this pipeline will search
for the output from the most recent run of genetree for the given catalog ID.</p>

</blockquote>
<img src="57.cluster.hist.png"/>
<blockquote><p>Distribution of the number of orthologs in each gene cluster.</p>
</blockquote>
<table class="table table-striped table-condensed table-mini">
<tr>
<td>Threshold</td>
<td># Clusters</td>
<td>% Missing Genes</td>
</tr>
<tr>
<td>3</td>
<td>12934</td>
<td>62.5%</td>
</tr>
<tr>
<td>4</td>
<td>11049</td>
<td>58.2%</td>
</tr>
<tr>
<td>5</td>
<td>9492</td>
<td>54.1%</td>
</tr>
<tr>
<td>6</td>
<td>8370</td>
<td>50.7%</td>
</tr>
<tr>
<td>7</td>
<td>7065</td>
<td>46.2%</td>
</tr>
<tr>
<td>8</td>
<td>5948</td>
<td>41.6%</td>
</tr>
<tr>
<td>9</td>
<td>5323</td>
<td>38.7%</td>
</tr>
<tr>
<td>10</td>
<td>4770</td>
<td>35.9%</td>
</tr>
<tr>
<td>11</td>
<td>4324</td>
<td>33.6%</td>
</tr>
<tr>
<td>12</td>
<td>3885</td>
<td>31.2%</td>
</tr>
<tr>
<td>13</td>
<td>3450</td>
<td>28.9%</td>
</tr>
<tr>
<td>14</td>
<td>3026</td>
<td>26.5%</td>
</tr>
<tr>
<td>15</td>
<td>2617</td>
<td>24.1%</td>
</tr>
<tr>
<td>16</td>
<td>2226</td>
<td>21.8%</td>
</tr>
<tr>
<td>17</td>
<td>1821</td>
<td>19.2%</td>
</tr>
<tr>
<td>18</td>
<td>1397</td>
<td>16.2%</td>
</tr>
<tr>
<td>19</td>
<td>1024</td>
<td>13.0%</td>
</tr>
<tr>
<td>20</td>
<td>760</td>
<td>10.3%</td>
</tr>
<tr>
<td>21</td>
<td>569</td>
<td>8.1%</td>
</tr>
<tr>
<td>22</td>
<td>382</td>
<td>6.0%</td>
</tr>
<tr>
<td>23</td>
<td>181</td>
<td>3.4%</td>
</tr>
<tr>
<td>24</td>
<td>34</td>
<td>0.0%</td>
</tr>
</table>
<blockquote></blockquote>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="multalign_58" class="stage">
<h2>multalign <small>(Run 58)</small></h2>
<blockquote>
<p>Applies sampling and length filters to each cluster of homologous sequences.</p>
<p>Creates multiple sequence alignments for each cluster of homologous sequences
using MAFFT using the E-INS-i algorithm; see doi: 10.1093/molbev/mst010.</p>
<p>Cleans up alignments using Gblocks.</p>

</blockquote>
<h4>Resourse Usage</h4>
<table class="table">
<tr>
<th>Wall Time (s)</th>
<th>User Time (s)</th>
<th>System Time (s)</th>
<th>Max Memory (KB)</th>
</tr>
<td class="right">47720.99 [sum]</td>
<td class="right">131016.28 [sum]</td>
<td class="right">65030.37 [sum]</td>
<td class="right">53,070,036 [max]</td>
</tr>
</table>
<a class="btn btn-info" onclick="togglestats(this)"><i class="icon-list-alt icon-white"></i> Show/hide details</a>
<table class="stats table table-striped table-condensed">
<tr><th>
Command</th><th>Stage</th><th>Wall Time (s)</th><th>User Time (s)</th><th>System Time (s)</th><th>Max Memory (KB)
</th></tr>
<tr>
<td>
parallel
</td>
<td>
align_sequences
</td>
<td class="right">
47683.88
</td>
<td class="right">
130845.83
</td>
<td class="right">
64930.54
</td>
<td class="right">
53,070,036
</td>
</tr>
<tr>
<td>
parallel
</td>
<td>
cleanup_alignments
</td>
<td class="right">
37.11
</td>
<td class="right">
170.45
</td>
<td class="right">
99.83
</td>
<td class="right">
508,788
</td>
</tr>
</tr>
</table>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="supermatrix_59" class="stage">
<h2>supermatrix <small>(Run 59)</small></h2>
<blockquote>
<p>Concatenates alignments of orhologous sequences to create a supermatrix.
It also creates a supermatrix with a given proportion of gene occupancy.</p>

</blockquote>
<p>Files: 
[<a href="59.supermatrix.fa">FASTA</a>] [<a href="59.supermatrix.partition.txt">partition</a>]
</p>
<table class="table table-striped table-condensed table-mini">
<tr>
<td>Sequence type</td>
<td style="text-align:right">nt</td>
</tr>
<tr>
<td>Target gene occupancy</td>
<td style="text-align:right">0.0%</td>
</tr>
<tr>
<td>Actual gene occupancy</td>
<td style="text-align:right">41.9%</td>
</tr>
<tr>
<td>Number of genes</td>
<td style="text-align:right">10,949</td>
</tr>
<tr>
<td>Number of columns</td>
<td style="text-align:right">12,758,237</td>
</tr>
</table>
<img src="59.supermatrix.png"/>
<blockquote><p>Image of the gene occupancy in the supermatrix, ordered by most
complete taxa, then most complete gene. Black indicates the gene is
present, white that it is absent.</p>
</blockquote>
<img src="59.supermatrix.cumalitve.png"/>
<blockquote><p>Plot of gene occupancies for cumulatively larger subsets of genes in
the supermatrix, ordered by most complete gene.</p>
</blockquote>
<table class="table table-striped table-condensed table-mini">
<tr>
<td>Species</td>
<td style="text-align:right">Number of genes</td>
<td style="text-align:right">Percent</td>
</tr>
<tr>
<td>Drosophila_virilis</td>
<td style="text-align:right">6850</td>
<td style="text-align:right">62.6%</td>
</tr>
<tr>
<td>Drosophila_mojavensis</td>
<td style="text-align:right">6813</td>
<td style="text-align:right">62.2%</td>
</tr>
<tr>
<td>Drosophila_willistoni</td>
<td style="text-align:right">6751</td>
<td style="text-align:right">61.7%</td>
</tr>
<tr>
<td>Drosophila_melanogaster</td>
<td style="text-align:right">6664</td>
<td style="text-align:right">60.9%</td>
</tr>
<tr>
<td>Drosophila_ananassae</td>
<td style="text-align:right">6662</td>
<td style="text-align:right">60.8%</td>
</tr>
<tr>
<td>Drosophila_pseudoobscura</td>
<td style="text-align:right">6647</td>
<td style="text-align:right">60.7%</td>
</tr>
<tr>
<td>Drosophila_grimshawi</td>
<td style="text-align:right">6501</td>
<td style="text-align:right">59.4%</td>
</tr>
<tr>
<td>Drosophila_nanella</td>
<td style="text-align:right">5349</td>
<td style="text-align:right">48.9%</td>
</tr>
<tr>
<td>Drosophila_cfdives</td>
<td style="text-align:right">5318</td>
<td style="text-align:right">48.6%</td>
</tr>
<tr>
<td>Scaptomyza_cyrtandrae</td>
<td style="text-align:right">4975</td>
<td style="text-align:right">45.4%</td>
</tr>
<tr>
<td>Scaptomyza_varipicta</td>
<td style="text-align:right">4838</td>
<td style="text-align:right">44.2%</td>
</tr>
<tr>
<td>Scaptomyza_varia</td>
<td style="text-align:right">4764</td>
<td style="text-align:right">43.5%</td>
</tr>
<tr>
<td>Drosophila_macrothrix</td>
<td style="text-align:right">4550</td>
<td style="text-align:right">41.6%</td>
</tr>
<tr>
<td>Drosophila_mimica</td>
<td style="text-align:right">4543</td>
<td style="text-align:right">41.5%</td>
</tr>
<tr>
<td>Drosophila_atroscutellata</td>
<td style="text-align:right">4385</td>
<td style="text-align:right">40.0%</td>
</tr>
<tr>
<td>Drosophila_primaeva</td>
<td style="text-align:right">3966</td>
<td style="text-align:right">36.2%</td>
</tr>
<tr>
<td>Drosophila_tanythrix</td>
<td style="text-align:right">3921</td>
<td style="text-align:right">35.8%</td>
</tr>
<tr>
<td>Drosophila_sproati</td>
<td style="text-align:right">3456</td>
<td style="text-align:right">31.6%</td>
</tr>
<tr>
<td>Drosophila_picticornis</td>
<td style="text-align:right">3115</td>
<td style="text-align:right">28.5%</td>
</tr>
<tr>
<td>Scaptomyza_hsui</td>
<td style="text-align:right">2299</td>
<td style="text-align:right">21.0%</td>
</tr>
<tr>
<td>Scaptomyza_graminum</td>
<td style="text-align:right">2252</td>
<td style="text-align:right">20.6%</td>
</tr>
<tr>
<td>Scaptomyza_montana</td>
<td style="text-align:right">2241</td>
<td style="text-align:right">20.5%</td>
</tr>
<tr>
<td>Drosophila_murphyi</td>
<td style="text-align:right">2228</td>
<td style="text-align:right">20.3%</td>
</tr>
<tr>
<td>Scaptomyza_pallida</td>
<td style="text-align:right">963</td>
<td style="text-align:right">8.8%</td>
</tr>
</table>
<blockquote><p>Number and percent of genes per taxon in supermatrix.</p>
</blockquote>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
<div id="supermatrix_60" class="stage">
<h2>supermatrix <small>(Run 60)</small></h2>
<blockquote>
<p>Concatenates alignments of orhologous sequences to create a supermatrix.
It also creates a supermatrix with a given proportion of gene occupancy.</p>

</blockquote>
<p>Files: 
[<a href="60.supermatrix.fa">FASTA</a>] [<a href="60.supermatrix.partition.txt">partition</a>]
</p>
<table class="table table-striped table-condensed table-mini">
<tr>
<td>Sequence type</td>
<td style="text-align:right">nt</td>
</tr>
<tr>
<td>Target gene occupancy</td>
<td style="text-align:right">80.0%</td>
</tr>
<tr>
<td>Actual gene occupancy</td>
<td style="text-align:right">80.0%</td>
</tr>
<tr>
<td>Number of genes</td>
<td style="text-align:right">1,926</td>
</tr>
<tr>
<td>Number of columns</td>
<td style="text-align:right">1,943,000</td>
</tr>
</table>
<img src="60.supermatrix.png"/>
<blockquote><p>Image of the gene occupancy in the supermatrix, ordered by most
complete taxa, then most complete gene. Black indicates the gene is
present, white that it is absent.</p>
</blockquote>
<img src="60.supermatrix.cumalitve.png"/>
<blockquote><p>Plot of gene occupancies for cumulatively larger subsets of genes in
the supermatrix, ordered by most complete gene.</p>
</blockquote>
<table class="table table-striped table-condensed table-mini">
<tr>
<td>Species</td>
<td style="text-align:right">Number of genes</td>
<td style="text-align:right">Percent</td>
</tr>
<tr>
<td>Drosophila_virilis</td>
<td style="text-align:right">1901</td>
<td style="text-align:right">98.7%</td>
</tr>
<tr>
<td>Drosophila_mojavensis</td>
<td style="text-align:right">1906</td>
<td style="text-align:right">99.0%</td>
</tr>
<tr>
<td>Drosophila_willistoni</td>
<td style="text-align:right">1898</td>
<td style="text-align:right">98.5%</td>
</tr>
<tr>
<td>Drosophila_melanogaster</td>
<td style="text-align:right">1895</td>
<td style="text-align:right">98.4%</td>
</tr>
<tr>
<td>Drosophila_ananassae</td>
<td style="text-align:right">1892</td>
<td style="text-align:right">98.2%</td>
</tr>
<tr>
<td>Drosophila_pseudoobscura</td>
<td style="text-align:right">1877</td>
<td style="text-align:right">97.5%</td>
</tr>
<tr>
<td>Drosophila_grimshawi</td>
<td style="text-align:right">1825</td>
<td style="text-align:right">94.8%</td>
</tr>
<tr>
<td>Drosophila_nanella</td>
<td style="text-align:right">1717</td>
<td style="text-align:right">89.1%</td>
</tr>
<tr>
<td>Drosophila_cfdives</td>
<td style="text-align:right">1686</td>
<td style="text-align:right">87.5%</td>
</tr>
<tr>
<td>Scaptomyza_cyrtandrae</td>
<td style="text-align:right">1769</td>
<td style="text-align:right">91.8%</td>
</tr>
<tr>
<td>Scaptomyza_varipicta</td>
<td style="text-align:right">1709</td>
<td style="text-align:right">88.7%</td>
</tr>
<tr>
<td>Scaptomyza_varia</td>
<td style="text-align:right">1761</td>
<td style="text-align:right">91.4%</td>
</tr>
<tr>
<td>Drosophila_macrothrix</td>
<td style="text-align:right">1587</td>
<td style="text-align:right">82.4%</td>
</tr>
<tr>
<td>Drosophila_mimica</td>
<td style="text-align:right">1595</td>
<td style="text-align:right">82.8%</td>
</tr>
<tr>
<td>Drosophila_atroscutellata</td>
<td style="text-align:right">1559</td>
<td style="text-align:right">80.9%</td>
</tr>
<tr>
<td>Drosophila_primaeva</td>
<td style="text-align:right">1346</td>
<td style="text-align:right">69.9%</td>
</tr>
<tr>
<td>Drosophila_tanythrix</td>
<td style="text-align:right">1450</td>
<td style="text-align:right">75.3%</td>
</tr>
<tr>
<td>Drosophila_sproati</td>
<td style="text-align:right">1352</td>
<td style="text-align:right">70.2%</td>
</tr>
<tr>
<td>Drosophila_picticornis</td>
<td style="text-align:right">1140</td>
<td style="text-align:right">59.2%</td>
</tr>
<tr>
<td>Scaptomyza_hsui</td>
<td style="text-align:right">1169</td>
<td style="text-align:right">60.7%</td>
</tr>
<tr>
<td>Scaptomyza_graminum</td>
<td style="text-align:right">1169</td>
<td style="text-align:right">60.7%</td>
</tr>
<tr>
<td>Scaptomyza_montana</td>
<td style="text-align:right">1167</td>
<td style="text-align:right">60.6%</td>
</tr>
<tr>
<td>Drosophila_murphyi</td>
<td style="text-align:right">1102</td>
<td style="text-align:right">57.2%</td>
</tr>
<tr>
<td>Scaptomyza_pallida</td>
<td style="text-align:right">510</td>
<td style="text-align:right">26.5%</td>
</tr>
</table>
<blockquote><p>Number and percent of genes per taxon in supermatrix.</p>
</blockquote>
<a class="btn" href="#toc"><i class="icon-list"></i> Back to TOC</a>
</div>
</body></html>